diff --git a/code/datums/ammo/rocket.dm b/code/datums/ammo/rocket.dm
index 4178e6b979b..f123d98cfb7 100644
--- a/code/datums/ammo/rocket.dm
+++ b/code/datums/ammo/rocket.dm
@@ -291,6 +291,52 @@
 
 	shell_speed = AMMO_SPEED_TIER_4
 
+//RIDGEWAY TANK CANNON
+/datum/ammo/rocket/ltb/ridgeway
+	name = "115mm round"
+	icon_state = "ridgeway"
+	flags_ammo_behavior = AMMO_EXPLOSIVE|AMMO_ROCKET
+
+	accuracy = HIT_ACCURACY_TIER_4
+	accurate_range = 32
+	max_range = 32
+	damage = 250
+	shell_speed = AMMO_SPEED_TIER_4
+	penetration= ARMOR_PENETRATION_TIER_6
+
+/datum/ammo/rocket/ltb/ridgeway/on_hit_mob(mob/mob, obj/projectile/projectile)
+	if(iscarbon(mob)) // Tank beats everything.
+		mob.ex_act(350, null, projectile.weapon_cause_data, 150)
+	if(mob.mob_size >= MOB_SIZE_BIG)
+		var/mob/living/alivent = mob
+		alivent.apply_armoured_damage(damage*15, ARMOR_BOMB, BRUTE, null, penetration)
+	cell_explosion(get_turf(mob), 175, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+	cell_explosion(get_turf(mob), 160, 100, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+
+/datum/ammo/rocket/ltb/ridgeway/on_hit_obj(obj/object, obj/projectile/projectile)
+	if(istype(object, /obj/vehicle/multitile))
+		var/obj/vehicle/multitile/mob = object
+		mob.next_move = world.time + vehicle_slowdown_time
+		playsound(mob, 'sound/effects/meteorimpact.ogg', 35)
+		mob.at_munition_interior_explosion_effect(cause_data = create_cause_data("Anti-Tank Rocket"))
+		mob.interior_crash_effect()
+		var/turf/turf = get_turf(mob.loc)
+		mob.ex_act(150, projectile.dir, projectile.weapon_cause_data, 100)
+		smoke.set_up(1, turf)
+		smoke.start()
+		return
+	cell_explosion(get_turf(object), 175, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+	cell_explosion(get_turf(object), 160, 100, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+
+/datum/ammo/rocket/ltb/ridgeway/on_hit_turf(turf/turf, obj/projectile/projectile)
+	cell_explosion(get_turf(turf), 175, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+	cell_explosion(get_turf(turf), 160, 100, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+
+/datum/ammo/rocket/ltb/ridgeway/do_at_max_range(obj/projectile/projectile)
+	cell_explosion(get_turf(projectile), 175, 50, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+	cell_explosion(get_turf(projectile), 160, 100, EXPLOSION_FALLOFF_SHAPE_LINEAR, null, projectile.weapon_cause_data)
+
+
 /datum/ammo/rocket/wp
 	name = "white phosphorous rocket"
 	flags_ammo_behavior = AMMO_ROCKET|AMMO_EXPLOSIVE|AMMO_STRIKES_SURFACE
diff --git a/code/datums/vehicles.dm b/code/datums/vehicles.dm
index 2611e7f5b57..a0e7468f58d 100644
--- a/code/datums/vehicles.dm
+++ b/code/datums/vehicles.dm
@@ -58,6 +58,10 @@
 	name = "UPP Command Tank"
 	interior_id = "upptank_command"
 
+/datum/map_template/interior/ridgeway
+	name = "Ridgeway Tank"
+	interior_id = "ridgeway"
+
 /datum/map_template/interior/aev
 	name = "AEV"
 	interior_id = "aev"
diff --git a/code/modules/projectiles/magazines/specialist.dm b/code/modules/projectiles/magazines/specialist.dm
index 1833d80bbcc..c794fcbee9e 100644
--- a/code/modules/projectiles/magazines/specialist.dm
+++ b/code/modules/projectiles/magazines/specialist.dm
@@ -535,6 +535,7 @@
 	default_ammo = /datum/ammo/energy/plasma
 	gun_type = /obj/item/weapon/gun/XM99
 
+
 // The "round-remover" power/fuel cell for the XM99
 /obj/item/ammo_magazine/plasma/gibber
 	name = "\improper XM99A1 lithium battery drum"
diff --git a/code/modules/vehicles/hardpoints/hardpoint_ammo/plasma_cannon.dm b/code/modules/vehicles/hardpoints/hardpoint_ammo/plasma_cannon.dm
new file mode 100644
index 00000000000..59f76510f59
--- /dev/null
+++ b/code/modules/vehicles/hardpoints/hardpoint_ammo/plasma_cannon.dm
@@ -0,0 +1,14 @@
+/obj/item/ammo_magazine/hardpoint/plasmacannon
+	name = "PARS 177 magazine"
+	desc = "A shielded cylindrical magazine for the Ridgeway Heavy Tank's point-defence plasma cannon, carries 3 2.4g Cadmium-Telluride pellets. Power is provided by a 7mW hydrogen fuel cell within the turret allows for 2500 firings."
+	icon = 'icons/obj/items/weapons/guns/ammo_by_faction/uscm.dmi'
+	caliber = "93mm" //Making this unique on purpose
+	icon_state = "plasmacannon_3"
+	w_class = SIZE_LARGE //Heavy
+	default_ammo = /datum/ammo/energy/plasma
+	max_rounds = 3
+	gun_type = /obj/item/hardpoint/primary/cannon/plasmacannon
+
+/obj/item/ammo_magazine/hardpoint/plasmacannon/update_icon()
+	icon_state = "plasmacannon_[current_rounds]"
+
diff --git a/code/modules/vehicles/hardpoints/hardpoint_ammo/ridgewaycannon_ammo.dm b/code/modules/vehicles/hardpoints/hardpoint_ammo/ridgewaycannon_ammo.dm
new file mode 100644
index 00000000000..194fdafe54f
--- /dev/null
+++ b/code/modules/vehicles/hardpoints/hardpoint_ammo/ridgewaycannon_ammo.dm
@@ -0,0 +1,12 @@
+/obj/item/ammo_magazine/hardpoint/ridgewaycannon
+	name = "115mm magazine"
+	desc = "  Magazine for the Ridgeway Heavy Tank's 115mm smoothbore cannon, feeding 6 shells into the autoloader. Lacking integrated propellant and relying on an internal hypergolic tank within the turret, these magazines benefit from high space efficiency and overall safety."
+	caliber = "115mm" //Making this unique on purpose
+	icon_state = "ridgewaycannon_6"
+	w_class = SIZE_LARGE //Heavy fucker
+	default_ammo = /datum/ammo/rocket/ltb/ridgeway
+	max_rounds = 6
+	gun_type = /obj/item/hardpoint/primary/cannon/ridgeway
+
+/obj/item/ammo_magazine/hardpoint/ridgewaycannon/update_icon()
+	icon_state = "ridgewaycannon_[current_rounds]"
diff --git a/code/modules/vehicles/hardpoints/holder/tank_turret.dm b/code/modules/vehicles/hardpoints/holder/tank_turret.dm
index e4c1105e091..427263146b9 100644
--- a/code/modules/vehicles/hardpoints/holder/tank_turret.dm
+++ b/code/modules/vehicles/hardpoints/holder/tank_turret.dm
@@ -303,3 +303,65 @@
 		data += list(H.get_tgui_info())
 
 	return data
+
+// Rideway turret
+/obj/item/hardpoint/holder/tank_turret/ridgeway
+	name = "\improper M40 Ridgeway Turret"
+	desc = "  The turret of the M40 Ridgeway heavy tank. A significant improvement over older vehicles, the Ridgeway is capable of deploying vastly superior weapons, defensive systems, optics and sensors without sacrificing on its armour or taking on excessive weight."
+
+	icon = 'icons/obj/vehicles/ridgeway.dmi'
+	icon_state = "tank_turret_0"
+	disp_icon = "tank"
+	disp_icon_state = "tank_turret"
+	activation_sounds = list('sound/weapons/vehicles/smokelauncher_fire.ogg')
+	pixel_x = -48
+	pixel_y = -48
+
+	density = TRUE //come on, it's huge
+
+	activatable = TRUE
+
+	ammo = new /obj/item/ammo_magazine/hardpoint/flare_launcher
+	max_clips = 5
+	use_muzzle_flash = FALSE
+
+	w_class = SIZE_MASSIVE
+	density = TRUE
+	anchored = TRUE
+
+	allowed_seat = VEHICLE_DRIVER
+
+	slot = HDPT_TURRET
+
+	// big beefy chonk of metal
+	health = 1500
+	damage_multiplier = 0.05
+
+	accepted_hardpoints = list(
+		// primaries
+        /obj/item/hardpoint/primary/cannon/plasmacannon,
+		/obj/item/hardpoint/primary/cannon/ridgeway,
+
+		// secondaries
+		/obj/item/hardpoint/secondary/small_flamer,
+		/obj/item/hardpoint/secondary/towlauncher,
+		/obj/item/hardpoint/secondary/m56cupola,
+		/obj/item/hardpoint/secondary/grenade_launcher
+	)
+
+	hdpt_layer = HDPT_LAYER_TURRET
+	px_offsets = list(
+		"1" = list(0, 0),
+		"2" = list(0, 24),
+		"4" = list(-20, 15),
+		"8" = list(20, 15)
+	)
+
+	scatter = 2
+	gun_firemode = GUN_FIREMODE_BURSTFIRE
+	gun_firemode_list = list(
+		GUN_FIREMODE_BURSTFIRE,
+	)
+	burst_amount = 2
+	burst_delay = 1.0 SECONDS
+	extra_delay = 5.0 SECONDS
diff --git a/code/modules/vehicles/hardpoints/primary/plasma_cannon.dm b/code/modules/vehicles/hardpoints/primary/plasma_cannon.dm
new file mode 100644
index 00000000000..882596b81b4
--- /dev/null
+++ b/code/modules/vehicles/hardpoints/primary/plasma_cannon.dm
@@ -0,0 +1,53 @@
+// Plasma cannon for ridgeway tank
+/obj/item/hardpoint/primary/cannon/plasmacannon
+	name = "\improper Boyars PARS 177 PDW"
+	desc = "The Boyars PARS 177 Point-Defence-Weapon is a 20kW phased plasma directed energy weapon used to defend the M40 Ridgeway from aerospace craft, missiles, and unofficially, visible hostile infantry. A welcome improvement over older laser based systems employed by the M22 Jackson and M34 Longstreet, providing hard-kill ability capable of downing aircraft and flash-vaporizing infantry."
+
+	icon = 'icons/obj/vehicles/hardpoints/tank.dmi'
+	icon_state = "plasma_cannon"
+	disp_icon = "ridgeway"
+	disp_icon_state = "plasma_cannon"
+	activation_sounds = list('sound/weapons/gun_xm99.ogg')
+
+	health = 2300
+	firing_arc = 60
+
+	ammo = new /obj/item/ammo_magazine/hardpoint/plasmacannon
+	max_clips = 3
+
+	px_offsets = list(
+		"1" = list(0, 46),
+		"2" = list(0, -43),
+		"4" = list(35, 2),
+		"8" = list(-35, 2)
+	)
+
+	use_muzzle_flash = FALSE
+	muzzleflash_icon_state = "muzzle_flash_blue"
+
+	muzzle_flash_pos = list(
+		"1" = list(0, 67),
+		"2" = list(-1, -88),
+		"4" = list(86, -6),
+		"8" = list(-86, -6)
+	)
+
+	scatter = 0
+	fire_delay = 10.0 SECONDS
+
+	var/obj/effect/ebeam/plasma_beam_type = /obj/effect/ebeam/laser/plasma
+	///world.time value, to prevent a lightshow without actually firing
+	var/beam_cooldown = 0
+	///Delay before another beam can start again, in tenths of seconds
+	var/beam_delay = 20
+
+/obj/item/hardpoint/primary/cannon/plasmacannon/handle_fire(atom/target, mob/living/user, /obj/vehicle/multitile/owner, params)
+
+    var/datum/beam/plasma_beam
+    if(!ammo.current_rounds)
+        click_empty(owner)
+        return
+    plasma_beam = target.beam(owner, "light_beam", 'icons/effects/beam.dmi', time = 0.7 SECONDS, maxdistance = 30, beam_type = plasma_beam_type, always_turn = TRUE)
+    animate(plasma_beam.visuals, alpha = 255, time = 0.7 SECONDS, color = COLOR_PURPLE, luminosity = 3 , easing = SINE_EASING|EASE_OUT)
+    . = ..()
+
diff --git a/code/modules/vehicles/hardpoints/primary/ridgeway_cannon.dm b/code/modules/vehicles/hardpoints/primary/ridgeway_cannon.dm
new file mode 100644
index 00000000000..068e8e80170
--- /dev/null
+++ b/code/modules/vehicles/hardpoints/primary/ridgeway_cannon.dm
@@ -0,0 +1,35 @@
+/obj/item/hardpoint/primary/cannon/ridgeway
+	name = "\improper M613 115mm Cannon"
+	desc = "  The main gun of the M40 Ridgeway heavy tank. A high-velocity smoothbore 115mm cannon capable of of a withering fire-rate of 60 rounds per minute. Paired with exceptional accuracy, it's able to combat any potential threat; from hostile armour, to reinforced bunkers. This cannon's autoloader seems to be damaged or modified, it probably won't achieve the same rate of fire, tough luck."
+
+	icon_state = "ridgeway_cannon"
+	disp_icon = "ridgeway"
+	disp_icon_state = "ridgeway_cannon"
+	activation_sounds = list('sound/weapons/vehicles/cannon_fire1.ogg', 'sound/weapons/vehicles/cannon_fire2.ogg')
+
+	health = 2000
+	firing_arc = 75
+
+	ammo = new /obj/item/ammo_magazine/hardpoint/ridgewaycannon
+	max_clips = 6
+
+	px_offsets = list(
+		"1" = list(0, 46),
+		"2" = list(0, -50),
+		"4" = list(61, 0),
+		"8" = list(-63, 0)
+	)
+
+	use_muzzle_flash = TRUE
+	angle_muzzleflash = FALSE
+	muzzleflash_icon_state = "muzzle_flash"
+
+	muzzle_flash_pos = list(
+		"1" = list(16, 100),
+		"2" = list(16, -110),
+		"4" = list(140, 4),
+		"8" = list(-105, 4)
+	)
+
+	scatter = 0
+	fire_delay = 3 SECONDS
diff --git a/code/modules/vehicles/hardpoints/wheels/treads.dm b/code/modules/vehicles/hardpoints/wheels/treads.dm
index ac774f2156f..a9e478322e3 100644
--- a/code/modules/vehicles/hardpoints/wheels/treads.dm
+++ b/code/modules/vehicles/hardpoints/wheels/treads.dm
@@ -46,3 +46,15 @@
 	owner.move_max_momentum = initial(owner.move_max_momentum)
 	owner.move_momentum_build_factor = initial(owner.move_momentum_build_factor)
 	owner.move_turn_momentum_loss_factor = initial(owner.move_turn_momentum_loss_factor)
+
+/obj/item/hardpoint/locomotion/treads/ridgeway
+	name = "\improper M40 Ridgeway Treads"
+	desc = "Treads for the M40 Ridgeway heavy tank. The least groundbreaking of the Ridgeway's advancements, though significant in their ability to tackle many of the problems faced by US forces in the colonies previous tanks couldn't. The tracks terrain adaptability and ease of maintenance design, makes it a favoured component for marine tankers"
+
+	icon_state = "treads_ridgeway"
+	disp_icon = "tank"
+	disp_icon_state = "treads_ridgeway"
+
+	health = 4500
+
+	move_max_momentum = 5 //same top speed, but takes 5 tiles to reach it
diff --git a/code/modules/vehicles/interior/areas.dm b/code/modules/vehicles/interior/areas.dm
index 2a9a6166c78..bf1cb135050 100644
--- a/code/modules/vehicles/interior/areas.dm
+++ b/code/modules/vehicles/interior/areas.dm
@@ -92,3 +92,7 @@
 /area/interior/vehicle/uppvan/prisoner
 	name = "\improper UPP van prisoner interior"
 	icon_state = "upp_van_prisoner"
+
+/area/interior/vehicle/uppvan/ridgeway
+	name = "\improper Ridgeway interior"
+	icon_state = "ridgeway"
diff --git a/code/modules/vehicles/interior/interactable/vehicle_locker.dm b/code/modules/vehicles/interior/interactable/vehicle_locker.dm
index fdbaf37d1f9..8e49bba11ed 100644
--- a/code/modules/vehicles/interior/interactable/vehicle_locker.dm
+++ b/code/modules/vehicles/interior/interactable/vehicle_locker.dm
@@ -130,6 +130,10 @@
 	icon = 'icons/obj/vehicles/interiors/tank.dmi'
 	icon_state = "locker"
 
+/obj/structure/vehicle_locker/tank/ridgeway
+	icon = 'icons/obj/vehicles/interiors/general.dmi'
+	icon_state = "rlocker"
+
 /obj/structure/vehicle_locker/tank/upp
 	icon = 'icons/obj/vehicles/interiors/upptank.dmi'
 	icon_state = "locker"
diff --git a/code/modules/vehicles/ridgewaytank/interior.dm b/code/modules/vehicles/ridgewaytank/interior.dm
new file mode 100644
index 00000000000..1672fc0ef66
--- /dev/null
+++ b/code/modules/vehicles/ridgewaytank/interior.dm
@@ -0,0 +1,7 @@
+/obj/structure/prop/vehicle/ridgeway
+	name = "Ridgeway Tank chassis"
+
+	icon = 'icons/obj/vehicles/interiors/ridgeway_chassis.dmi'
+	icon_state = "ridgeway_chassis"
+	layer = ABOVE_TURF_LAYER
+	mouse_opacity = FALSE
diff --git a/code/modules/vehicles/ridgewaytank/ridgewaytank.dm b/code/modules/vehicles/ridgewaytank/ridgewaytank.dm
new file mode 100644
index 00000000000..3672598f807
--- /dev/null
+++ b/code/modules/vehicles/ridgewaytank/ridgewaytank.dm
@@ -0,0 +1,172 @@
+// Ridgeway Tank
+/obj/vehicle/multitile/tank/ridgeway
+	name = "M40 Ridgeway Heavy Tank"
+	desc = "A giant piece of armor with a big gun, you know what to do. Entrance in the back."
+	desc_lore = "The M40 Ridgeway heavy tank is the premier armoured vehicle in use by the United States armed forces. An advanced piece of equipment introduced in the late 2170s, the Ridgeway is the epitome of a modern fighting vehicle; featuring extensive automation and advanced defensive systems including a independent-targeting plasma hard-kill phalanx and cutting edge stealth capabilities making it effectively invisible to most basic ground sensors. Despite being slated to replace the outdated M22 Jackson, it has seen limited deployment among Colonial Marine units."
+
+	icon = 'icons/obj/vehicles/ridgeway.dmi'
+	icon_state = "tank_base"
+	pixel_x = -65
+	pixel_y = -65
+
+	bound_width = 96
+	bound_height = 96
+
+	bound_x = -32
+	bound_y = -32
+
+	interior_map = /datum/map_template/interior/ridgeway
+
+	passengers_slots = 4
+	//this is done in case VCs die inside the tank, so that someone else can come in and take them out.
+	revivable_dead_slots = 2
+	xenos_slots = 4
+
+	entrances = list(
+		"back" = list(0, 2)
+	)
+
+	movement_sound = 'sound/vehicles/tank_driving.ogg'
+	honk_sound = 'sound/vehicles/honk_3_ambulence.ogg'
+
+	required_skill = SKILL_VEHICLE_LARGE
+
+	vehicle_flags = VEHICLE_CLASS_MEDIUM
+
+	move_max_momentum = 2
+	move_momentum_build_factor = 1.5
+	move_turn_momentum_loss_factor = 0.6
+
+	light_range = 5
+
+	wall_ram_damage = 350
+	vehicle_ram_multiplier = 20 //Big heavy chunk of metal cares not for crunchies
+
+	// Rest (all the guns) is handled by the tank turret hardpoint
+	hardpoints_allowed = list(
+		/obj/item/hardpoint/holder/tank_turret/ridgeway,
+		/obj/item/hardpoint/support/weapons_sensor,
+		/obj/item/hardpoint/support/overdrive_enhancer,
+		/obj/item/hardpoint/support/artillery_module,
+		/obj/item/hardpoint/armor/snowplow,
+		/obj/item/hardpoint/locomotion/treads/ridgeway,
+	)
+
+	seats = list(
+		VEHICLE_DRIVER = null,
+		VEHICLE_GUNNER = null,
+	)
+
+	active_hp = list(
+		VEHICLE_DRIVER = null,
+		VEHICLE_GUNNER = null,
+	)
+
+	dmg_multipliers = list(
+		"all" = 1,
+		"acid" = 3.5,
+		"slash" = 0.3,
+		"bullet" = 0.4,
+		"explosive" = 0.6,
+		"blunt" = 0.3,
+		"abstract" = 1,
+	)
+
+	explosive_resistance = 1000
+
+/obj/vehicle/multitile/tank/ridgeway/initialize_cameras(change_tag = FALSE)
+	if(!camera)
+		camera = new /obj/structure/machinery/camera/vehicle(src)
+	if(change_tag)
+		camera.c_tag = "#[rand(1,100)] M40 \"[nickname]\" Heavy Tank" //this fluff allows it to be at the start of cams list M40 Ridgeway Heavy Tank
+		if(camera_int)
+			camera_int.c_tag = camera.c_tag + " interior" //this fluff allows it to be at the start of cams list
+	else
+		camera.c_tag = "#[rand(1,100)] M40 Heavy Tank"
+		if(camera_int)
+			camera_int.c_tag = camera.c_tag + " interior" //this fluff allows it to be at the start of cams list
+
+/obj/vehicle/multitile/tank/ridgeway/load_hardpoints()
+	add_hardpoint(new /obj/item/hardpoint/holder/tank_turret/ridgeway)
+
+// ** PRESETS SPAWNERS
+
+/obj/effect/vehicle_spawner/ridgeway
+
+	name = "Tank Spawner"
+	icon = 'icons/obj/vehicles/ridgeway.dmi'
+	icon_state = "tank_base"
+	pixel_x = -48
+	pixel_y = -48
+
+/obj/effect/vehicle_spawner/ridgeway/Initialize()
+	. = ..()
+	spawn_vehicle()
+	qdel(src)
+
+//PRESET: turret, no hardpoints (not the one without turret for convenience, you still expect to have turret when you spawn "no hardpoints tank")
+/obj/effect/vehicle_spawner/ridgeway/spawn_vehicle()
+	var/obj/vehicle/multitile/tank/ridgeway/TANK = new (loc)
+
+	load_misc(TANK)
+	load_hardpoints(TANK)
+	handle_direction(TANK)
+	TANK.update_icon()
+
+	return TANK
+
+/obj/effect/vehicle_spawner/ridgeway/load_hardpoints(obj/vehicle/multitile/tank/ridgeway/vic)
+	vic.add_hardpoint(new /obj/item/hardpoint/holder/tank_turret/ridgeway)
+
+//PRESET: turret, treads installed
+/obj/effect/vehicle_spawner/ridgeway/plain/load_hardpoints(obj/vehicle/multitile/tank/ridgeway/vic)
+	vic.add_hardpoint(new /obj/item/hardpoint/holder/tank_turret/ridgeway)
+	vic.add_hardpoint(new /obj/item/hardpoint/locomotion/treads)
+
+//PRESET: no hardpoints
+/obj/effect/vehicle_spawner/ridgeway/hull/load_hardpoints(obj/vehicle/multitile/tank/ridgeway/vic)
+	return
+
+//Just the hull and it's broken TOO, you get the full experience
+/obj/effect/vehicle_spawner/ridgeway/hull/broken/spawn_vehicle()
+	var/obj/vehicle/multitile/tank/ridgeway/tonk = ..()
+	load_damage(tonk)
+	tonk.update_icon()
+
+//PRESET: default hardpoints, destroyed
+/obj/effect/vehicle_spawner/ridgeway/decrepit/spawn_vehicle()
+	var/obj/vehicle/multitile/tank/ridgeway/TANK = new (loc)
+
+	load_misc(TANK)
+	handle_direction(TANK)
+	load_hardpoints(TANK)
+	load_damage(TANK)
+	TANK.update_icon()
+
+/obj/effect/vehicle_spawner/ridgeway/decrepit/load_hardpoints(obj/vehicle/multitile/tank/ridgeway/vic)
+	vic.add_hardpoint(new /obj/item/hardpoint/support/weapons_sensor)
+	vic.add_hardpoint(new /obj/item/hardpoint/locomotion/treads/ridgeway)
+	vic.add_hardpoint(new /obj/item/hardpoint/holder/tank_turret/ridgeway)
+	for(var/obj/item/hardpoint/holder/tank_turret/ridgeway/tonkturret in vic.hardpoints)
+		tonkturret.add_hardpoint(new /obj/item/hardpoint/primary/cannon/ridgeway)
+		break
+
+//PRESET: default hardpoints
+/obj/effect/vehicle_spawner/ridgeway/fixed/load_hardpoints(obj/vehicle/multitile/tank/ridgeway/vic)
+	vic.add_hardpoint(new /obj/item/hardpoint/support/weapons_sensor)
+	vic.add_hardpoint(new /obj/item/hardpoint/locomotion/treads/ridgeway)
+	vic.add_hardpoint(new /obj/item/hardpoint/holder/tank_turret/ridgeway)
+	for(var/obj/item/hardpoint/holder/tank_turret/ridgeway/tonkturret in vic.hardpoints)
+		tonkturret.add_hardpoint(new /obj/item/hardpoint/primary/cannon/ridgeway)
+		tonkturret.add_hardpoint(new /obj/item/hardpoint/secondary/m56cupola)
+		break
+
+//PRESET: Plasma hardpoints
+/obj/effect/vehicle_spawner/ridgeway/plasma/load_hardpoints(obj/vehicle/multitile/tank/ridgeway/vic)
+	vic.add_hardpoint(new /obj/item/hardpoint/support/artillery_module)
+	vic.add_hardpoint(new /obj/item/hardpoint/locomotion/treads/ridgeway)
+	vic.add_hardpoint(new /obj/item/hardpoint/holder/tank_turret/ridgeway)
+	for(var/obj/item/hardpoint/holder/tank_turret/ridgeway/tonkturret in vic.hardpoints)
+		tonkturret.add_hardpoint(new /obj/item/hardpoint/primary/cannon/plasmacannon)
+		tonkturret.add_hardpoint(new /obj/item/hardpoint/secondary/small_flamer)
+		break
diff --git a/colonialmarines.dme b/colonialmarines.dme
index e2b70b1e956..cc9aa79b777 100644
--- a/colonialmarines.dme
+++ b/colonialmarines.dme
@@ -2621,8 +2621,10 @@
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\ltb_ammo.dm"
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\minigun_ammo.dm"
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\p17702.dm"
+#include "code\modules\vehicles\hardpoints\hardpoint_ammo\plasma_cannon.dm"
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\primary_flamer_ammo.dm"
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\railgun.dm"
+#include "code\modules\vehicles\hardpoints\hardpoint_ammo\ridgewaycannon_ammo.dm"
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\secondary_flamer_ammo.dm"
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\smoke_ammo.dm"
 #include "code\modules\vehicles\hardpoints\hardpoint_ammo\t60p3m.dm"
@@ -2641,8 +2643,10 @@
 #include "code\modules\vehicles\hardpoints\primary\ltb.dm"
 #include "code\modules\vehicles\hardpoints\primary\minigun.dm"
 #include "code\modules\vehicles\hardpoints\primary\p17702.dm"
+#include "code\modules\vehicles\hardpoints\primary\plasma_cannon.dm"
 #include "code\modules\vehicles\hardpoints\primary\primary.dm"
 #include "code\modules\vehicles\hardpoints\primary\railgun.dm"
+#include "code\modules\vehicles\hardpoints\primary\ridgeway_cannon.dm"
 #include "code\modules\vehicles\hardpoints\primary\van_mg.dm"
 #include "code\modules\vehicles\hardpoints\secondary\aa_launcher.dm"
 #include "code\modules\vehicles\hardpoints\secondary\cupola.dm"
@@ -2683,6 +2687,8 @@
 #include "code\modules\vehicles\multitile\multitile_interaction.dm"
 #include "code\modules\vehicles\multitile\multitile_movement.dm"
 #include "code\modules\vehicles\multitile\multitile_verbs.dm"
+#include "code\modules\vehicles\ridgewaytank\interior.dm"
+#include "code\modules\vehicles\ridgewaytank\ridgewaytank.dm"
 #include "code\modules\vehicles\tank\aev.dm"
 #include "code\modules\vehicles\tank\interior.dm"
 #include "code\modules\vehicles\tank\tank.dm"
diff --git a/icons/obj/items/weapons/guns/ammo_by_faction/uscm.dmi b/icons/obj/items/weapons/guns/ammo_by_faction/uscm.dmi
index a79e6e8ac79..ee09cf3ec8f 100644
Binary files a/icons/obj/items/weapons/guns/ammo_by_faction/uscm.dmi and b/icons/obj/items/weapons/guns/ammo_by_faction/uscm.dmi differ
diff --git a/icons/obj/items/weapons/projectiles.dmi b/icons/obj/items/weapons/projectiles.dmi
index a8f20eda786..2cc64dc0a29 100644
Binary files a/icons/obj/items/weapons/projectiles.dmi and b/icons/obj/items/weapons/projectiles.dmi differ
diff --git a/icons/obj/vehicles/hardpoints/tank.dmi b/icons/obj/vehicles/hardpoints/tank.dmi
index a8c49e5bddd..3adfd750b2d 100644
Binary files a/icons/obj/vehicles/hardpoints/tank.dmi and b/icons/obj/vehicles/hardpoints/tank.dmi differ
diff --git a/icons/obj/vehicles/interiors/general.dmi b/icons/obj/vehicles/interiors/general.dmi
index 9e13720e201..1ea1d832617 100644
Binary files a/icons/obj/vehicles/interiors/general.dmi and b/icons/obj/vehicles/interiors/general.dmi differ
diff --git a/icons/obj/vehicles/interiors/ridgeway_chassis.dmi b/icons/obj/vehicles/interiors/ridgeway_chassis.dmi
new file mode 100644
index 00000000000..c5f1ec59b36
Binary files /dev/null and b/icons/obj/vehicles/interiors/ridgeway_chassis.dmi differ
diff --git a/icons/obj/vehicles/ridgeway.dmi b/icons/obj/vehicles/ridgeway.dmi
new file mode 100644
index 00000000000..3e054a60c37
Binary files /dev/null and b/icons/obj/vehicles/ridgeway.dmi differ
diff --git a/icons/turf/areas_interiors.dmi b/icons/turf/areas_interiors.dmi
index 17dbc795332..9d22702945e 100644
Binary files a/icons/turf/areas_interiors.dmi and b/icons/turf/areas_interiors.dmi differ
diff --git a/maps/interiors/ridgeway.dmm b/maps/interiors/ridgeway.dmm
new file mode 100644
index 00000000000..bd075b4c803
--- /dev/null
+++ b/maps/interiors/ridgeway.dmm
@@ -0,0 +1,113 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
+"a" = (
+/obj/structure/prop/vehicle/ridgeway{
+	pixel_x = 16;
+	pixel_y = -16
+	},
+/turf/open/void/vehicle,
+/area/space)
+"c" = (
+/turf/open/void/vehicle,
+/area/space)
+"h" = (
+/obj/effect/landmark/interior/spawn/vehicle_gunner_seat/armor{
+	dir = 4
+	},
+/obj/effect/landmark/interior/spawn/weapons_loader{
+	pixel_y = 24;
+	icon_state = "weapons_loader_ridgeway"
+	},
+/obj/structure/vehicle_locker/tank/ridgeway{
+	pixel_x = -23;
+	pixel_y = 9
+	},
+/turf/open/shuttle/vehicle/floor_3_11,
+/area/interior/vehicle/uppvan/ridgeway)
+"o" = (
+/obj/effect/landmark/interior/spawn/vehicle_commander_seat/armor{
+	dir = 4
+	},
+/turf/open/shuttle/vehicle/floor_3_12,
+/area/interior/vehicle/uppvan/ridgeway)
+"r" = (
+/obj/structure/phone_base{
+	name = "Tank Telephone";
+	phone_id = "Cheetah 2A Light Tank";
+	phone_category = "Vehicles";
+	dir = 4;
+	pixel_x = -14;
+	pixel_y = -25
+	},
+/turf/open/shuttle/vehicle/floor_1_14,
+/area/interior/vehicle/uppvan/ridgeway)
+"x" = (
+/obj/effect/landmark/interior/spawn/entrance{
+	dir = 8;
+	exit_type = /obj/structure/interior_exit/vehicle/tank;
+	name = "back entrance marker";
+	tag = "back"
+	},
+/obj/structure/extinguisher_cabinet/lifeboat{
+	icon = 'icons/obj/vehicles/interiors/general.dmi';
+	pixel_x = 10;
+	pixel_y = 22;
+	layer = 3.1
+	},
+/obj/structure/machinery/cm_vending/sorted/medical/wall_med/vehicle{
+	pixel_y = 22;
+	pixel_x = -1
+	},
+/turf/open/shuttle/vehicle/floor_1_13,
+/area/interior/vehicle/uppvan/ridgeway)
+"C" = (
+/turf/open/void/vehicle/unopacity,
+/area/interior/vehicle/uppvan/ridgeway)
+"O" = (
+/obj/effect/landmark/interior/spawn/vehicle_driver_seat/armor{
+	dir = 4
+	},
+/turf/open/shuttle/vehicle/floor_3_5,
+/area/interior/vehicle/uppvan/ridgeway)
+
+(1,1,1) = {"
+c
+c
+c
+a
+c
+"}
+(2,1,1) = {"
+c
+C
+x
+C
+c
+"}
+(3,1,1) = {"
+c
+h
+r
+o
+c
+"}
+(4,1,1) = {"
+c
+C
+O
+C
+c
+"}
+(5,1,1) = {"
+c
+C
+C
+C
+c
+"}
+(6,1,1) = {"
+c
+c
+c
+c
+c
+"}
