diff --git a/code/datums/vehicles.dm b/code/datums/vehicles.dm
index 2611e7f5b57..931e0f1009a 100644
--- a/code/datums/vehicles.dm
+++ b/code/datums/vehicles.dm
@@ -30,6 +30,10 @@
 	name = "Movie APC"
 	interior_id = "apc_movie"
 
+/datum/map_template/interior/apc_wolfpack
+	name = "Wolfpack APC"
+	interior_id = "apc_wolfpack"
+
 /datum/map_template/interior/apc_aa
 	name = "Anti-Air APC"
 	interior_id = "apc_aa"
diff --git a/code/modules/vehicles/apc/apc_wolfpack.dm b/code/modules/vehicles/apc/apc_wolfpack.dm
new file mode 100644
index 00000000000..ad51d867518
--- /dev/null
+++ b/code/modules/vehicles/apc/apc_wolfpack.dm
@@ -0,0 +1,204 @@
+
+/obj/vehicle/multitile/apc/wolfpack
+	name = "M577A3E2 Mobile Gun System"
+	desc = "An M577A3E2 Mobile Gun System complete with a tactical operations center. Entrance on the right."
+
+	icon = 'icons/obj/vehicles/apc_wolfpack.dmi'
+	icon_state = "apc_base_wolfpack"
+	pixel_x = -64
+	pixel_y = -64
+
+	bound_width = 96
+	bound_height = 96
+
+	interior_map = /datum/map_template/interior/apc_wolfpack
+
+
+	passengers_slots = 20
+
+	entrances = list(
+		"right rear" = list(-2, 1),
+		"right front" = list(-2, 0)
+	)
+
+	seats = list(
+		VEHICLE_DRIVER = null,
+		VEHICLE_GUNNER = null,
+	)
+
+	active_hp = list(
+		VEHICLE_DRIVER = null,
+		VEHICLE_GUNNER = null,
+	)
+	dmg_multipliers = list(
+		"all" = 1,
+		"acid" = 2.6,
+		"slash" = 0.5,
+		"bullet" = 0.7,
+		"explosive" = 0.9,
+		"blunt" = 0.5,
+		"abstract" = 1,
+	)
+
+	wall_ram_damage = 100
+	vehicle_ram_multiplier = 10
+	hardpoints_allowed = list(
+		/obj/item/hardpoint/holder/tank_turret/wolfpack,
+		/obj/item/hardpoint/secondary/frontalcannon,
+		/obj/item/hardpoint/support/artillery_module,
+		/obj/item/hardpoint/locomotion/apc_wheels,
+	)
+
+	move_max_momentum = 2.3
+	move_momentum_build_factor = 1.5
+	move_turn_momentum_loss_factor = 0.7
+
+/obj/vehicle/multitile/apc/wolfpack/add_seated_verbs(mob/living/M, seat)
+	if(!M.client)
+		return
+	add_verb(M.client, list(
+		/obj/vehicle/multitile/proc/get_status_info,
+		/obj/vehicle/multitile/proc/open_controls_guide,
+		/obj/vehicle/multitile/proc/name_vehicle,
+	))
+	if(seat == VEHICLE_DRIVER)
+		add_verb(M.client, list(
+			/obj/vehicle/multitile/proc/toggle_door_lock,
+			/obj/vehicle/multitile/proc/activate_horn,
+		))
+	else if(seat == VEHICLE_GUNNER)
+		add_verb(M.client, list(
+			/obj/vehicle/multitile/proc/cycle_hardpoint,
+			/obj/vehicle/multitile/proc/toggle_gyrostabilizer,
+			/obj/vehicle/multitile/proc/switch_hardpoint,
+		))
+
+/obj/vehicle/multitile/apc/wolfpack/remove_seated_verbs(mob/living/M, seat)
+	if(!M.client)
+		return
+	remove_verb(M.client, list(
+		/obj/vehicle/multitile/proc/get_status_info,
+		/obj/vehicle/multitile/proc/open_controls_guide,
+		/obj/vehicle/multitile/proc/name_vehicle,
+	))
+	SStgui.close_user_uis(M, src)
+	if(seat == VEHICLE_DRIVER)
+		remove_verb(M.client, list(
+			/obj/vehicle/multitile/proc/toggle_door_lock,
+			/obj/vehicle/multitile/proc/activate_horn,
+		))
+	else if(seat == VEHICLE_GUNNER)
+		remove_verb(,.client, list(
+			/obj/vehicle/multitile/proc/cycle_hardpoint,
+			/obj/vehicle/multitile/proc/toggle_gyrostabilizer,
+			/obj/vehicle/multitile/proc/switch_hardpoint,
+		))
+
+/obj/vehicle/multitile/apc/wolfpack/relaymove(mob/user, direction)
+	if(user == seats[VEHICLE_DRIVER])
+		return ..()
+
+	if(user != seats[VEHICLE_GUNNER])
+		return FALSE
+
+	var/obj/item/hardpoint/holder/tank_turret/wolfpack/turret = null
+	for(var/obj/item/hardpoint/holder/tank_turret/wolfpack/tonkturret in hardpoints)
+		turret = tonkturret
+		break
+	if(!turret)
+		return FALSE
+
+	if(direction == GLOB.reverse_dir[turret.dir] || direction == turret.dir)
+		return FALSE
+
+	turret.user_rotation(user, turning_angle(turret.dir, direction))
+	update_icon()
+
+	return TRUE
+
+/obj/vehicle/multitile/apc/wolfpack/MouseDrop_T(mob/dropped, mob/user)
+	. = ..()
+	if((dropped != user) || !isxeno(user))
+		return
+
+	if(health > 0)
+		to_chat(user, SPAN_XENO("We can't jump over [src] until it is destroyed!"))
+		return
+
+	var/turf/current_turf = get_turf(user)
+	var/dir_to_go = get_dir(current_turf, src)
+	for(var/i in 1 to 3)
+		current_turf = get_step(current_turf, dir_to_go)
+		if(!(current_turf in locs))
+			break
+
+		if(current_turf.density)
+			to_chat(user, SPAN_XENO("The path over [src] is obstructed!"))
+			return
+
+/obj/vehicle/multitile/apc/wolfpack/initialize_cameras(change_tag = FALSE)
+	if(!camera)
+		camera = new /obj/structure/machinery/camera/vehicle(src)
+	if(change_tag)
+		camera.c_tag = "#[rand(1,100)] M577A3 \"[nickname]\" APC"
+		if(camera_int)
+			camera_int.c_tag = camera.c_tag + " interior"
+	else
+		camera.c_tag = "#[rand(1,100)] M577A3 APC"
+		if(camera_int)
+			camera_int.c_tag = camera.c_tag + " interior"
+
+/obj/vehicle/multitile/apc/wolfpack/set_muzzle_offsets(obj/item/hardpoint/HP)
+	//sets muzzle flash offsets for APC weapons as appropriate for the movie APC
+	switch(HP.slot)
+		if(HDPT_PRIMARY) //dualcannon
+			HP.muzzle_flash_pos = list(
+				"1" = list(8, -12),
+				"2" = list(25, 24),
+				"4" = list(0, 27),
+				"8" = list(30, 9)
+			)
+		if(HDPT_SECONDARY) //frontalcannon
+			HP.muzzle_flash_pos = list(
+				"1" = list(34, 76),
+				"2" = list(-1, -76),
+				"4" = list(94, -7),
+				"8" = list(-62, 12)
+			)
+		if(HDPT_SUPPORT) //flare
+			HP.muzzle_flash_pos = list(
+				"1" = list(22, -13),
+				"2" = list(11, 22),
+				"4" = list(2, 10),
+				"8" = list(30, 24)
+			)
+	return ..()
+
+/obj/effect/vehicle_spawner/apc_wolfpack
+	name = "Wolfpack APC Spawner"
+	icon = 'icons/obj/vehicles/apc_wolfpack.dmi'
+	icon_state = "apc_base_wolfpack"
+	pixel_x = -64
+	pixel_y = -64
+
+/obj/effect/vehicle_spawner/apc_wolfpack/Initialize()
+	. = ..()
+	spawn_vehicle()
+	qdel(src)
+
+/obj/effect/vehicle_spawner/apc_wolfpack/spawn_vehicle()
+	var/obj/vehicle/multitile/apc/wolfpack/APC = new (loc)
+
+	load_misc(APC)
+	load_hardpoints(APC)
+	handle_direction(APC)
+	APC.update_icon()
+
+//PRESET: only wheels installed
+/obj/effect/vehicle_spawner/apc_wolfpack/plain/load_hardpoints(obj/vehicle/multitile/apc/movie/V)
+	V.add_hardpoint(new /obj/item/hardpoint/holder/tank_turret/wolfpack)
+	V.add_hardpoint(new /obj/item/hardpoint/secondary/frontalcannon)
+	V.add_hardpoint(new /obj/item/hardpoint/locomotion/apc_wheels)
+	V.add_hardpoint(new /obj/item/hardpoint/support/artillery_module)
+	for(var/obj/item/hardpoint/holder/tank_turret/wolfpack/tonkturret in V.hardpoints)
+		tonkturret.add_hardpoint(new /obj/item/hardpoint/primary/cannon/wolfpack)
diff --git a/code/modules/vehicles/hardpoints/holder/tank_turret.dm b/code/modules/vehicles/hardpoints/holder/tank_turret.dm
index e4c1105e091..4052f3d460e 100644
--- a/code/modules/vehicles/hardpoints/holder/tank_turret.dm
+++ b/code/modules/vehicles/hardpoints/holder/tank_turret.dm
@@ -303,3 +303,12 @@
 		data += list(H.get_tgui_info())
 
 	return data
+
+
+/obj/item/hardpoint/holder/tank_turret/wolfpack
+	accepted_hardpoints = list(
+		// primaries
+		/obj/item/hardpoint/primary/cannon/wolfpack,
+		// secondaries
+	)
+	icon = 'icons/obj/vehicles/apc_wolfpack.dmi'
diff --git a/code/modules/vehicles/hardpoints/primary/ltb.dm b/code/modules/vehicles/hardpoints/primary/ltb.dm
index 5a1090df546..5d3b78e83dc 100644
--- a/code/modules/vehicles/hardpoints/primary/ltb.dm
+++ b/code/modules/vehicles/hardpoints/primary/ltb.dm
@@ -36,3 +36,18 @@
 		BULLET_TRAIT_ENTRY(/datum/element/bullet_trait_iff),
 		BULLET_TRAIT_ENTRY_ID("vehicles", /datum/element/bullet_trait_damage_boost, 300, GLOB.damage_boost_vehicles),
 	))
+
+/obj/item/hardpoint/primary/cannon/wolfpack
+	px_offsets = list(
+		"1" = list(0, 0),
+		"2" = list(0, 0),
+		"4" = list(0, 0),
+		"8" = list(0, 0)
+	)
+
+	muzzle_flash_pos = list(
+		"1" = list(0, 65),
+		"2" = list(0, -80),
+		"4" = list(95, -4),
+		"8" = list(-95, -4)
+	)
diff --git a/colonialmarines.dme b/colonialmarines.dme
index e2b70b1e956..5c0240b543d 100644
--- a/colonialmarines.dme
+++ b/colonialmarines.dme
@@ -2596,6 +2596,7 @@
 #include "code\modules\vehicles\apc\apc_medical.dm"
 #include "code\modules\vehicles\apc\apc_movie.dm"
 #include "code\modules\vehicles\apc\apc_moviemed.dm"
+#include "code\modules\vehicles\apc\apc_wolfpack.dm"
 #include "code\modules\vehicles\apc\arc.dm"
 #include "code\modules\vehicles\apc\arc_weaponless.dm"
 #include "code\modules\vehicles\apc\interior.dm"
diff --git a/icons/obj/structures/props/ice_colony/fabs_greebles.dmi b/icons/obj/structures/props/ice_colony/fabs_greebles.dmi
index a367d1d11c9..46151aec599 100644
Binary files a/icons/obj/structures/props/ice_colony/fabs_greebles.dmi and b/icons/obj/structures/props/ice_colony/fabs_greebles.dmi differ
diff --git a/icons/obj/vehicles/apc_wolfpack.dmi b/icons/obj/vehicles/apc_wolfpack.dmi
new file mode 100644
index 00000000000..f6951441e0d
Binary files /dev/null and b/icons/obj/vehicles/apc_wolfpack.dmi differ
diff --git a/icons/obj/vehicles/interiors/apc.dmi b/icons/obj/vehicles/interiors/apc.dmi
index f34d30f03d0..b95cf2889e2 100644
Binary files a/icons/obj/vehicles/interiors/apc.dmi and b/icons/obj/vehicles/interiors/apc.dmi differ
diff --git a/maps/interiors/apc_wolfpack.dmm b/maps/interiors/apc_wolfpack.dmm
new file mode 100644
index 00000000000..6cfcf29583c
--- /dev/null
+++ b/maps/interiors/apc_wolfpack.dmm
@@ -0,0 +1,589 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
+"a" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "rear_6"
+	},
+/turf/open/void/vehicle,
+/area/space)
+"b" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wall";
+	pixel_y = -12;
+	opacity = 0
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = -4;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/obj/structure/prop/invuln/ice_prefab/roof_greeble{
+	icon_state = "vent6";
+	pixel_x = -2;
+	pixel_y = -11
+	},
+/turf/open/void/vehicle,
+/area/space)
+"c" = (
+/obj/structure/machinery/computer/overwatch/toc{
+	layer = 2.97;
+	pixel_y = 12
+	},
+/obj/structure/surface/table/reinforced/toc/west{
+	pixel_y = -7
+	},
+/obj/structure/machinery/computer/shuttle/dropship/flight/toc{
+	pixel_y = 12;
+	pixel_x = 7
+	},
+/obj/item/toy/handcard/aceofspades{
+	pixel_x = -8;
+	pixel_y = -3
+	},
+/turf/open/shuttle/vehicle/floor_1_3,
+/area/interior/vehicle/apc/movie)
+"d" = (
+/obj/structure/bed/chair/vehicle{
+	dir = 1;
+	pixel_x = 8
+	},
+/obj/structure/bed/chair/vehicle{
+	dir = 1;
+	pixel_x = -8
+	},
+/obj/effect/landmark/interior/spawn/interior_viewport{
+	dir = 1;
+	pixel_x = 2;
+	pixel_y = -13
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"e" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = 4;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/turf/open/void/vehicle,
+/area/space)
+"f" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_1";
+	pixel_x = 1;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/turf/open/void/vehicle,
+/area/space)
+"g" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "front_1"
+	},
+/turf/open/void/vehicle,
+/area/space)
+"h" = (
+/obj/structure/bed/chair/vehicle{
+	dir = 1;
+	pixel_x = -8
+	},
+/obj/structure/bed/chair/vehicle{
+	dir = 1;
+	pixel_x = 8
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"i" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wall_tally";
+	opacity = 0
+	},
+/turf/open/space/basic,
+/area/space)
+"j" = (
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"l" = (
+/obj/structure/machinery/prop/almayer/CICmap{
+	density = 0;
+	icon_state = "toc_map";
+	layer = 2.97;
+	name = "Tactical Map Display";
+	pixel_y = 11;
+	pixel_x = -7
+	},
+/obj/structure/surface/table/reinforced/toc/east{
+	pixel_y = -7
+	},
+/obj/structure/phone_base{
+	name = "APC reciever";
+	phone_category = "Vehicles";
+	phone_id = "M577A3 Armoured Personnel Carrier";
+	pixel_y = 11;
+	pixel_x = 11
+	},
+/turf/open/shuttle/vehicle/floor_1_4,
+/area/interior/vehicle/apc/movie)
+"n" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	layer = 2.01;
+	pixel_y = -8
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "corner_smaller_R";
+	opacity = 0;
+	pixel_y = 12
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	layer = 2.01
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "intersection_1";
+	opacity = 0
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "corner_small_L";
+	opacity = 0;
+	pixel_y = 11
+	},
+/turf/open/space/basic,
+/area/space)
+"o" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "rear_1"
+	},
+/turf/open/void/vehicle,
+/area/space)
+"p" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wall_2";
+	opacity = 0
+	},
+/turf/open/space/basic,
+/area/space)
+"q" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "front_wheel_L_1"
+	},
+/turf/open/space/basic,
+/area/space)
+"r" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "rear_wheel_R";
+	layer = 5.2
+	},
+/turf/open/space/basic,
+/area/space)
+"t" = (
+/obj/structure/bed/chair/vehicle/toc{
+	dir = 1;
+	pixel_y = 19;
+	pixel_x = 16
+	},
+/turf/open/shuttle/vehicle/floor_1_3,
+/area/interior/vehicle/apc/movie)
+"v" = (
+/obj/structure/vehicle_locker/movie{
+	pixel_y = 31;
+	layer = 2.9
+	},
+/obj/effect/landmark/interior/spawn/weapons_loader,
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"w" = (
+/obj/effect/landmark/interior/spawn/interior_camera{
+	dir = 5;
+	pixel_x = -18;
+	pixel_y = 38
+	},
+/obj/structure/vehicle_locker/movie{
+	pixel_y = 11;
+	layer = 2.9
+	},
+/obj/structure/gun_rack/apc/frontal{
+	pixel_y = 1;
+	pixel_x = 7;
+	density = 0
+	},
+/obj/structure/gun_rack/apc/frontal{
+	pixel_y = 1;
+	pixel_x = -8;
+	density = 0
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"x" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wall";
+	opacity = 0;
+	pixel_y = -12
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = 10;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = -18;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/turf/open/void/vehicle,
+/area/space)
+"y" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "front_5"
+	},
+/turf/open/void/vehicle,
+/area/space)
+"z" = (
+/obj/effect/landmark/interior/spawn/entrance{
+	alpha = 50;
+	exit_type = /obj/structure/interior_exit/vehicle/apc;
+	name = "Right APC door";
+	tag = "right rear"
+	},
+/turf/open/shuttle/vehicle/floor_1_3,
+/area/interior/vehicle/apc/movie)
+"A" = (
+/obj/effect/landmark/interior/spawn/vehicle_gunner_seat/armor{
+	dir = 1
+	},
+/obj/structure/prop/almayer/computers/sensor_computer1{
+	pixel_y = 26;
+	density = 0;
+	name = "turret fire control system";
+	desc = "The IBM series 10 computer retrofitted to work as a fire control system for the M577A3E1 MGS. While somewhat dated it still serves its purpose."
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"B" = (
+/turf/open/shuttle/vehicle/floor_1_3,
+/area/interior/vehicle/apc/movie)
+"C" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "door_big_right";
+	layer = 5.2;
+	pixel_y = 32;
+	alpha = 100
+	},
+/turf/open/void/vehicle,
+/area/space)
+"D" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "rear_5"
+	},
+/turf/open/void/vehicle,
+/area/space)
+"E" = (
+/turf/open/void/vehicle,
+/area/space)
+"F" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "intersection_2";
+	opacity = 0
+	},
+/obj/effect/landmark/interior/spawn/interior_viewport{
+	pixel_x = 6;
+	pixel_y = -10
+	},
+/obj/structure/prop/ice_colony/hula_girl{
+	pixel_y = 9;
+	pixel_x = -12;
+	layer = 3.5
+	},
+/obj/item/ashtray/plastic{
+	pixel_y = 6;
+	pixel_x = 10
+	},
+/obj/item/trash/cigbutt{
+	pixel_y = 13;
+	pixel_x = 10
+	},
+/obj/item/tool/pen/red/clicky{
+	pixel_y = 8;
+	pixel_x = 2
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"G" = (
+/obj/structure/bed/chair/vehicle{
+	pixel_x = -8;
+	pixel_y = 14
+	},
+/obj/structure/machinery/cm_vending/sorted/medical/wall_med/vehicle{
+	pixel_y = 30;
+	pixel_x = -8
+	},
+/obj/structure/bed/chair/vehicle{
+	pixel_x = 8;
+	pixel_y = 14
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"H" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "corner_inverse_R_1";
+	opacity = 0
+	},
+/obj/structure/sign/poster/pinup{
+	pixel_x = 7;
+	layer = 2.5
+	},
+/turf/open/space/basic,
+/area/space)
+"I" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "front_3"
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "front_4";
+	pixel_y = 12
+	},
+/turf/open/void/vehicle,
+/area/space)
+"J" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = 4;
+	pixel_y = 1;
+	layer = 2.01;
+	opacity = 0
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "corner_smaller_R";
+	opacity = 0;
+	pixel_y = 12
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "corner_smaller_L";
+	opacity = 0;
+	pixel_y = 12
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "intersection_2";
+	opacity = 0;
+	pixel_y = -20
+	},
+/obj/item/ammo_box/magazine/mk1/large/ap/empty{
+	pixel_y = 3
+	},
+/obj/item/toy/deck{
+	pixel_x = 6;
+	pixel_y = -1
+	},
+/turf/open/shuttle/vehicle,
+/area/space)
+"K" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "corner_smaller_L";
+	opacity = 0
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_back_top_2";
+	pixel_x = -4;
+	pixel_y = -4
+	},
+/turf/open/void/vehicle,
+/area/space)
+"L" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "rear_wheel_L"
+	},
+/turf/open/space/basic,
+/area/space)
+"M" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "rear_3"
+	},
+/turf/open/void/vehicle,
+/area/space)
+"N" = (
+/turf/open/shuttle/vehicle/floor_1_4,
+/area/interior/vehicle/apc/movie)
+"O" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "front_2";
+	pixel_y = 7
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "front_2";
+	pixel_y = -2;
+	layer = 2.01
+	},
+/turf/open/void/vehicle,
+/area/space)
+"P" = (
+/obj/structure/barricade/handrail{
+	dir = 8
+	},
+/obj/effect/landmark/interior/spawn/interior_viewport{
+	dir = 8;
+	pixel_x = 5
+	},
+/obj/structure/bed/chair/vehicle{
+	dir = 4
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"Q" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "door_big_left";
+	pixel_y = 32;
+	layer = 5.2;
+	alpha = 100
+	},
+/turf/open/void/vehicle,
+/area/space)
+"R" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wall_wheelside"
+	},
+/turf/open/space/basic,
+/area/space)
+"S" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "front_6";
+	pixel_y = -1
+	},
+/turf/open/void/vehicle,
+/area/space)
+"T" = (
+/obj/effect/landmark/interior/spawn/entrance{
+	alpha = 50;
+	exit_type = /obj/structure/interior_exit/vehicle/apc;
+	name = "Right APC door";
+	tag = "right front"
+	},
+/turf/open/shuttle/vehicle/floor_1_4,
+/area/interior/vehicle/apc/movie)
+"U" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "front_wheel_R";
+	layer = 4.1
+	},
+/turf/open/space/basic,
+/area/space)
+"V" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_back_top_1";
+	pixel_x = -4;
+	pixel_y = -4
+	},
+/turf/open/void/vehicle,
+/area/space)
+"W" = (
+/obj/effect/landmark/interior/spawn/interior_viewport{
+	dir = 8;
+	pixel_x = 5;
+	pixel_y = 18
+	},
+/obj/effect/landmark/interior/spawn/vehicle_driver_seat/armor/movie{
+	dir = 4;
+	buckling_y = 16;
+	pixel_y = 16
+	},
+/turf/open/shuttle/vehicle/floor_2,
+/area/interior/vehicle/apc/movie)
+"Y" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "rear_2"
+	},
+/turf/open/void/vehicle,
+/area/space)
+"Z" = (
+/obj/structure/interior_wall/apc{
+	icon_state = "corner_smaller_R";
+	opacity = 0
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = 4;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/obj/structure/interior_wall/apc{
+	icon_state = "wheel_front_top_2";
+	pixel_x = 8;
+	pixel_y = -5;
+	layer = 2.01
+	},
+/turf/open/void/vehicle,
+/area/space)
+
+(1,1,1) = {"
+a
+D
+M
+Y
+o
+E
+"}
+(2,1,1) = {"
+V
+i
+v
+j
+L
+E
+"}
+(3,1,1) = {"
+K
+n
+F
+A
+r
+E
+"}
+(4,1,1) = {"
+b
+c
+t
+B
+z
+Q
+"}
+(5,1,1) = {"
+x
+l
+N
+N
+T
+C
+"}
+(6,1,1) = {"
+Z
+H
+G
+h
+R
+E
+"}
+(7,1,1) = {"
+e
+J
+w
+d
+q
+E
+"}
+(8,1,1) = {"
+f
+p
+W
+P
+U
+E
+"}
+(9,1,1) = {"
+S
+I
+O
+y
+g
+E
+"}
