diff --git a/code/__DEFINES/typecheck/xenos.dm b/code/__DEFINES/typecheck/xenos.dm
index 34b70ac92f4..0c2cdec5435 100644
--- a/code/__DEFINES/typecheck/xenos.dm
+++ b/code/__DEFINES/typecheck/xenos.dm
@@ -23,8 +23,13 @@
 #define isspitter(A) (istype(A, /mob/living/carbon/xenomorph/spitter))
 #define iswarrior(A) (istype(A, /mob/living/carbon/xenomorph/warrior))
 #define isburrower(A) (istype(A, /mob/living/carbon/xenomorph/burrower))
+#define isspiderguard(A) (istype(A, /mob/living/carbon/xenomorph/spider_guard))
+#define isspidernurse(A) (istype(A, /mob/living/carbon/xenomorph/spider_nurse))
+#define isspiderhunter(A) (istype(A, /mob/living/carbon/xenomorph/spider_hunter))
+#define islizard(A) (istype(A, /mob/living/carbon/xenomorph/giant_lizard))
 
 #define isxeno_builder(A) (isdrone(A) || ishivelord(A) || iscarrier(A) || isburrower(A) || isqueen(A))
+#define isspider(A) (isspiderguard(A) || isspidernurse(A) || isspiderhunter(A))
 
 /// Returns true/false based on if the xenomorph can harm the passed carbon mob.
 /mob/living/carbon/xenomorph/proc/can_not_harm(mob/living/carbon/attempt_harm_mob)
diff --git a/code/__DEFINES/xeno.dm b/code/__DEFINES/xeno.dm
index b1a2545361e..a075b776033 100644
--- a/code/__DEFINES/xeno.dm
+++ b/code/__DEFINES/xeno.dm
@@ -240,6 +240,7 @@
 // Health bands
 #define XENO_HEALTH_LARVA 45 * XENO_UNIVERSAL_HPMULT
 #define XENO_HEALTH_LESSER_DRONE 160 * XENO_UNIVERSAL_HPMULT
+#define XENO_HEALTH_SPIDER 200 * XENO_UNIVERSAL_HPMULT
 #define XENO_HEALTH_RUNNER 230 * XENO_UNIVERSAL_HPMULT // Killed by 1 PB
 #define XENO_HEALTH_TIER_1 250 * XENO_UNIVERSAL_HPMULT
 #define XENO_HEALTH_TIER_2 300 * XENO_UNIVERSAL_HPMULT
@@ -323,6 +324,8 @@
 #define XENO_SPEED_TIER_6 -0.6
 #define XENO_SPEED_TIER_7 -0.8
 #define XENO_SPEED_HELLHOUND -1
+#define XENO_SPEED_SPIDER -1.1
+#define XENO_SPEED_FASTSPIDER -1.1
 #define XENO_SPEED_TIER_8 -1.4
 #define XENO_SPEED_TIER_9 -1.6
 #define XENO_SPEED_TIER_10 -1.8
@@ -704,6 +707,10 @@
 #define XENO_CASTE_QUEEN  "Queen"
 #define XENO_CASTE_PREDALIEN  "Predalien"
 #define XENO_CASTE_HELLHOUND  "Hellhound"
+#define XENO_CASTE_SPIDER_GUARD "Spider Guard"
+#define XENO_CASTE_SPIDER_NURSE "Spider Nurse"
+#define XENO_CASTE_SPIDER_HUNTER "Spider Hunter"
+#define XENO_CASTE_GIANT_LIZARD "Giant Lizard"
 #define XENO_SPECIAL_CASTES   list(XENO_CASTE_QUEEN, XENO_CASTE_PREDALIEN, XENO_CASTE_HELLHOUND)
 
 #define ALL_XENO_CASTES list(XENO_CASTE_LARVA, XENO_CASTE_PREDALIEN_LARVA, XENO_CASTE_FACEHUGGER, XENO_CASTE_LESSER_DRONE, XENO_CASTE_DRONE, XENO_CASTE_RUNNER, XENO_CASTE_SENTINEL, XENO_CASTE_DEFENDER, XENO_CASTE_BURROWER, XENO_CASTE_CARRIER, XENO_CASTE_HIVELORD, XENO_CASTE_LURKER, XENO_CASTE_WARRIOR, XENO_CASTE_SPITTER, XENO_CASTE_BOILER, XENO_CASTE_PRAETORIAN, XENO_CASTE_CRUSHER, XENO_CASTE_RAVAGER, XENO_CASTE_QUEEN, XENO_CASTE_PREDALIEN, XENO_CASTE_HELLHOUND, XENO_CASTE_SOLDIER, XENO_CASTE_KING)
diff --git a/code/game/jobs/role_authority.dm b/code/game/jobs/role_authority.dm
index 3cf4c46db49..c15569aa653 100644
--- a/code/game/jobs/role_authority.dm
+++ b/code/game/jobs/role_authority.dm
@@ -750,6 +750,14 @@ I hope it's easier to tell what the heck this proc is even doing, unlike previou
 			M = /mob/living/carbon/xenomorph/predalien
 		if(XENO_CASTE_HELLHOUND)
 			M = /mob/living/carbon/xenomorph/hellhound
+		if(XENO_CASTE_SPIDER_GUARD)
+			M = /mob/living/carbon/xenomorph/spider_guard
+		if(XENO_CASTE_SPIDER_NURSE)
+			M = /mob/living/carbon/xenomorph/spider_nurse
+		if(XENO_CASTE_SPIDER_HUNTER)
+			M = /mob/living/carbon/xenomorph/spider_hunter
+		if(XENO_CASTE_GIANT_LIZARD)
+			M = /mob/living/carbon/xenomorph/giant_lizard
 		if(XENO_CASTE_SOLDIER)
 			M = /mob/living/carbon/xenomorph/soldier
 		if(XENO_CASTE_KING)
diff --git a/code/game/sound.dm b/code/game/sound.dm
index 6fd371423ef..3d17ab026f3 100644
--- a/code/game/sound.dm
+++ b/code/game/sound.dm
@@ -476,6 +476,8 @@
 				sound = pick('sound/voice/larva_talk1.ogg','sound/voice/larva_talk2.ogg','sound/voice/larva_talk3.ogg','sound/voice/larva_talk4.ogg')
 			if("hiss_talk")
 				sound = pick('sound/voice/hiss2.ogg','sound/voice/hiss3.ogg','sound/voice/hiss4.ogg')
+			if("click_talk")
+				sound = pick('sound/voice/pred_click1.ogg','sound/voice/pred_click2.ogg','sound/voice/pred_click3.ogg','sound/voice/pred_click4.ogg','sound/voice/pred_click5.ogg')
 			if("alien_growl")
 				sound = pick('sound/voice/alien_growl1.ogg','sound/voice/alien_growl2.ogg','sound/voice/alien_growl3.ogg')
 			if("alien_hiss")
diff --git a/code/modules/admin/game_master/game_master.dm b/code/modules/admin/game_master/game_master.dm
index 4d3d24abafd..e68e574a930 100644
--- a/code/modules/admin/game_master/game_master.dm
+++ b/code/modules/admin/game_master/game_master.dm
@@ -42,8 +42,8 @@ GLOBAL_VAR_INIT(radio_communication_clarity, 100)
 	message_admins("[src] has [MODE_HAS_TOGGLEABLE_FLAG(MODE_NO_MAKE_BARRICADES) ? "disabled" : "enabled"] Barricade Creation.")
 
 // Spawn stuff
-#define DEFAULT_SPAWN_XENO_STRING XENO_CASTE_DRONE
-#define GAME_MASTER_AI_XENOS list(XENO_CASTE_DRONE, XENO_CASTE_SOLDIER, XENO_CASTE_RUNNER, RUNNER_ACIDER, XENO_CASTE_LURKER, XENO_CASTE_CRUSHER, XENO_CASTE_FACEHUGGER)
+#define DEFAULT_SPAWN_XENO_STRING XENO_CASTE_SPIDER_GUARD
+#define GAME_MASTER_AI_XENOS list(XENO_CASTE_SPIDER_GUARD, XENO_CASTE_SPIDER_NURSE, XENO_CASTE_SPIDER_HUNTER, XENO_CASTE_GIANT_LIZARD, XENO_CASTE_DRONE, XENO_CASTE_SOLDIER, XENO_CASTE_RUNNER, RUNNER_ACIDER, XENO_CASTE_LURKER, XENO_CASTE_CRUSHER, XENO_CASTE_FACEHUGGER)
 #define DEFAULT_SPAWN_HIVE_STRING XENO_HIVE_NORMAL
 
 #define DEFAULT_XENO_AMOUNT_TO_SPAWN 1
diff --git a/code/modules/admin/game_master/game_master_submenu/ambush.dm b/code/modules/admin/game_master/game_master_submenu/ambush.dm
index 09ad5d2389a..598dc01fa5a 100644
--- a/code/modules/admin/game_master/game_master_submenu/ambush.dm
+++ b/code/modules/admin/game_master/game_master_submenu/ambush.dm
@@ -1,6 +1,6 @@
 
-#define DEFAULT_SPAWN_XENO_STRING XENO_CASTE_DRONE
-#define GAME_MASTER_AMBUSH_AI_XENOS list(XENO_CASTE_DRONE, XENO_CASTE_SOLDIER, XENO_CASTE_RUNNER, XENO_CASTE_LURKER, XENO_CASTE_FACEHUGGER)
+#define DEFAULT_SPAWN_XENO_STRING XENO_CASTE_SPIDER_GUARD
+#define GAME_MASTER_AMBUSH_AI_XENOS list(XENO_CASTE_SPIDER_GUARD, XENO_CASTE_SPIDER_NURSE, XENO_CASTE_SPIDER_HUNTER, XENO_CASTE_DRONE, XENO_CASTE_SOLDIER, XENO_CASTE_RUNNER, XENO_CASTE_LURKER, XENO_CASTE_FACEHUGGER)
 #define DEFAULT_SPAWN_HIVE_STRING XENO_HIVE_NORMAL
 
 #define DEFAULT_XENO_AMOUNT_TO_SPAWN 1
diff --git a/code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm b/code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm
index 4fe2c5c29db..c8086a831a3 100644
--- a/code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm
+++ b/code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm
@@ -29,6 +29,18 @@
 	var/plant_on_semiweedable = FALSE
 	var/node_type = /obj/effect/alien/weeds/node
 
+/datum/action/xeno_action/onclick/plant_webs
+	name = "Plant Webs (10)"
+	ability_name = "Plant Webs"
+	plasma_cost = 10
+	macro_path = /datum/action/xeno_action/verb/verb_plant_webs
+	action_type = XENO_ACTION_CLICK
+	xeno_cooldown = 2
+	ability_primacy = XENO_PRIMARY_ACTION_1
+
+	var/plant_on_semiweedable = FALSE
+	var/node_type = /obj/effect/spider/stickyweb
+
 // Resting
 /datum/action/xeno_action/onclick/xeno_resting
 	name = "Rest"
diff --git a/code/modules/mob/living/carbon/xenomorph/abilities/general_ability_macros.dm b/code/modules/mob/living/carbon/xenomorph/abilities/general_ability_macros.dm
index 5e22064cfeb..4e4fa661698 100644
--- a/code/modules/mob/living/carbon/xenomorph/abilities/general_ability_macros.dm
+++ b/code/modules/mob/living/carbon/xenomorph/abilities/general_ability_macros.dm
@@ -16,6 +16,13 @@
 	var/action_name = "Plant Weeds (75)"
 	handle_xeno_macro(src, action_name)
 
+/datum/action/xeno_action/verb/verb_plant_webs()
+	set category = "Alien"
+	set name = "Plant Webs"
+	set hidden = TRUE
+	var/action_name = "Plant Weeds (10)"
+	handle_xeno_macro(src, action_name)
+
 /datum/action/xeno_action/verb/verb_mark_resin()
 	set category = "Alien"
 	set name = "Mark Resin"
diff --git a/code/modules/mob/living/carbon/xenomorph/ai/movement/spider.dm b/code/modules/mob/living/carbon/xenomorph/ai/movement/spider.dm
new file mode 100644
index 00000000000..b7c7e90eaff
--- /dev/null
+++ b/code/modules/mob/living/carbon/xenomorph/ai/movement/spider.dm
@@ -0,0 +1,111 @@
+#define BLACKLIST_TURF_TIMEOUT (180 SECONDS)
+
+/datum/xeno_ai_movement/spider
+	var/last_home_turf
+	var/list/blacklisted_turfs = list()
+
+/datum/xeno_ai_movement/spider/Destroy(force, ...)
+	. = ..()
+	blacklisted_turfs = null
+
+//spiders expand the web clusters
+/datum/xeno_ai_movement/spider/ai_move_idle(delta_time)
+	if(!GLOB.ai_xeno_weeding)
+		return ..()
+
+	var/mob/living/carbon/xenomorph/idle_xeno = parent
+
+	if(idle_xeno.throwing)
+		return
+
+	if(home_turf)
+		if(get_dist(home_turf, idle_xeno) > max_distance_from_home)
+			home_turf = null
+			return
+
+		if(get_dist(home_turf, idle_xeno) <= 0)
+			var/datum/action/xeno_action/onclick/plant_webs/plant_webs_action = get_action(parent, /datum/action/xeno_action/onclick/plant_webs)
+			INVOKE_ASYNC(plant_webs_action, TYPE_PROC_REF(/datum/action/xeno_action/onclick/plant_webs, use_ability_wrapper))
+			home_turf = null
+			return
+
+		if(!idle_xeno.move_to_next_turf(home_turf, home_locate_range))
+			home_turf = null
+			return
+
+		return
+
+	if(next_home_search > world.time)
+		return
+
+	var/turf/current_turf = get_turf(idle_xeno)
+	next_home_search = world.time + home_search_delay
+
+	var/shortest_distance
+	for(var/turf/potential_home as anything in shuffle(RANGE_TURFS(home_locate_range, current_turf)))
+		if(!check_turf(potential_home))
+			continue
+
+		if(!isnull(shortest_distance) && get_dist(idle_xeno, potential_home) > shortest_distance)
+			continue
+
+		shortest_distance = get_dist(idle_xeno, potential_home)
+		home_turf = potential_home
+
+	if(!home_turf)
+		idle_xeno.set_resting(TRUE, FALSE, TRUE)
+		return
+
+	idle_xeno.set_resting(FALSE, FALSE, TRUE)
+
+	if(home_turf == last_home_turf)
+		blacklisted_turfs += home_turf
+		addtimer(CALLBACK(src, PROC_REF(unblacklist_turf), home_turf), BLACKLIST_TURF_TIMEOUT)
+
+	last_home_turf = home_turf
+
+/datum/xeno_ai_movement/spider/proc/unblacklist_turf(turf/unblacklisting_turf)
+	blacklisted_turfs -= unblacklisting_turf
+
+/datum/xeno_ai_movement/spider/proc/check_turf(turf/checked_turf)
+	var/area/found_area = get_area(checked_turf)
+	if(found_area.flags_area & AREA_NOTUNNEL)
+		return FALSE
+
+	if(found_area.flags_area & AREA_UNWEEDABLE)
+		return FALSE
+
+	if(!found_area.can_build_special)
+		return FALSE
+
+	if(checked_turf in blacklisted_turfs)
+		return FALSE
+
+	var/obj/effect/alien/weeds/checked_weeds = checked_turf.weeds
+	if(checked_weeds && IS_SAME_HIVENUMBER(checked_weeds, parent))
+		return FALSE
+
+	if(checked_turf.is_weedable() < FULLY_WEEDABLE)
+		return FALSE
+
+	var/obj/effect/alien/weeds/found_weeds = locate(/obj/effect/alien/weeds/node) in range(3, checked_turf)
+	if(found_weeds && IS_SAME_HIVENUMBER(found_weeds, parent))
+		return FALSE
+
+	if(checked_turf.density)
+		return FALSE
+
+	var/blocked = FALSE
+	for(var/atom/potential_blocker as anything in checked_turf)
+		if(parent != potential_blocker && (potential_blocker.density || potential_blocker.can_block_movement))
+			blocked = TRUE
+			break
+
+	if(blocked)
+		return FALSE
+
+	for(var/obj/structure/struct in checked_turf)
+		if(struct.density && !(struct.flags_atom & ON_BORDER))
+			return FALSE
+
+	return TRUE
diff --git a/code/modules/mob/living/carbon/xenomorph/castes/Giant_Lizard.dm b/code/modules/mob/living/carbon/xenomorph/castes/Giant_Lizard.dm
new file mode 100644
index 00000000000..d813bb6b324
--- /dev/null
+++ b/code/modules/mob/living/carbon/xenomorph/castes/Giant_Lizard.dm
@@ -0,0 +1,88 @@
+/datum/caste_datum/giant_lizard //The jungle nightmare
+	caste_type = XENO_CASTE_GIANT_LIZARD //For when you want to hunt folks with lizards
+	tier = 1
+	melee_damage_lower = XENO_DAMAGE_TIER_3
+	melee_damage_upper = XENO_DAMAGE_TIER_4
+	melee_vehicle_damage = XENO_DAMAGE_TIER_2
+	max_health = XENO_HEALTH_TIER_3
+	plasma_gain = XENO_PLASMA_GAIN_TIER_8
+	plasma_max = XENO_PLASMA_TIER_10
+	xeno_explosion_resistance = XENO_NO_EXPLOSIVE_ARMOR
+	armor_deflection = XENO_NO_ARMOR
+	evasion = XENO_EVASION_LOW
+	speed = XENO_SPEED_HELLHOUND
+	fire_vulnerability_mult = FIRE_MULTIPLIER_EXTREME //weak to fire
+
+	caste_desc = "A prowler of the jungles."
+	evolves_to = list()
+	deevolves_to = list()
+
+	heal_resting = 1
+	heal_standing = 0.5
+	heal_knocked_out = 0.25
+	innate_healing = TRUE
+
+	behavior_delegate_type = /datum/behavior_delegate/giant_lizard
+	minimap_icon = "runner"
+
+/mob/living/carbon/xenomorph/giant_lizard
+	caste_type = XENO_CASTE_GIANT_LIZARD
+	name = XENO_CASTE_GIANT_LIZARD
+	desc = "A large, wolf-like reptile. Its eyes are keenly focused on yours."
+	icon = 'icons/mob/xenos/giant_lizard.dmi'
+	icon_size = 64
+	pixel_x = -16
+	old_x = -16
+	icon_state = "Normal Giant Lizard Running"
+	plasma_types = list(PLASMA_CHITIN)
+	mob_size = MOB_SIZE_XENO_SMALL
+	tier = 1
+
+	slash_verb = "bite"
+	slashes_verb = "bites"
+	slash_sound = 'sound/weapons/bite.ogg'
+
+	acid_blood_damage = 0 /// it's a lizard innit
+	acid_blood_spatter = FALSE /// We dont want that for these guys
+
+	icon_xeno = 'icons/mob/xenos/giant_lizard.dmi'
+	icon_xenonid = 'icons/mob/xenos/giant_lizard.dmi'
+
+	weed_food_icon = 'icons/mob/xenos/weeds.dmi'
+	weed_food_states = list("Hellhound_1","Hellhound_2","Hellhound_3")
+	weed_food_states_flipped = list("Hellhound_1","Hellhound_2","Hellhound_3")
+
+	base_actions = list(
+		/datum/action/xeno_action/onclick/xeno_resting,
+		/datum/action/xeno_action/activable/pounce,
+	)
+	inherent_verbs = list(
+		/mob/living/carbon/xenomorph/proc/vent_crawl,
+	)
+	gib_chance = 0
+	claw_type = CLAW_TYPE_SHARP
+	pull_multiplier = 0.8
+
+	target_unconscious = FALSE
+
+/datum/action/xeno_action/activable/pounce
+		ai_prob_chance = 50
+		distance = 3
+
+/datum/behavior_delegate/giant_lizard/melee_attack_additional_effects_target(mob/living/carbon/A)
+	if(prob(50))
+		A.apply_damage(5, OXY)
+		to_chat(A, SPAN_XENOHIGHDANGER("You feel woozy, as the [bound_xeno] bites into you with teeth that drip with shimmering saliva!"))
+		A.sway_jitter(times = 5, steps = 3)
+		A.apply_effect(1, PARALYZE)
+		A.apply_effect(10, EYE_BLUR)
+
+
+/mob/living/carbon/xenomorph/giant_lizard/death(cause, gibbed)
+	. = ..(cause, gibbed, "lets out a waning growl.")
+	if(!.)
+		return
+
+/mob/living/carbon/xenomorph/giant_lizard/handle_blood_splatter(splatter_dir)
+	new /obj/effect/temp_visual/dir_setting/bloodsplatter/human(loc, splatter_dir)
+
diff --git a/code/modules/mob/living/carbon/xenomorph/castes/Spider_Guard.dm b/code/modules/mob/living/carbon/xenomorph/castes/Spider_Guard.dm
new file mode 100644
index 00000000000..57829cfff2b
--- /dev/null
+++ b/code/modules/mob/living/carbon/xenomorph/castes/Spider_Guard.dm
@@ -0,0 +1,85 @@
+/datum/caste_datum/spider_guard //The basic spider enemy
+	caste_type = XENO_CASTE_SPIDER_GUARD //For when you want to send a horde of spiders
+	tier = 1
+	melee_damage_lower = XENO_DAMAGE_TIER_2
+	melee_damage_upper = XENO_DAMAGE_TIER_3
+	melee_vehicle_damage = XENO_DAMAGE_TIER_2
+	max_health = XENO_HEALTH_SPIDER
+	plasma_gain = XENO_PLASMA_GAIN_TIER_8
+	plasma_max = XENO_PLASMA_TIER_10
+	xeno_explosion_resistance = XENO_EXPLOSIVE_ARMOR_TIER_1
+	armor_deflection = XENO_ARMOR_TIER_1
+	evasion = XENO_EVASION_LOW
+	speed = XENO_SPEED_SPIDER
+	fire_vulnerability_mult = FIRE_MULTIPLIER_DEADLY //v weak to fire
+
+	caste_desc = "A guard of the cluster."
+	evolves_to = list(XENO_CASTE_SPIDER_NURSE, XENO_CASTE_SPIDER_HUNTER)
+	deevolves_to = list()
+
+	tackle_min = 1
+	tackle_max = 3
+	tackle_chance = 10
+
+	heal_resting = 2
+	heal_standing = 1
+	heal_knocked_out = 1.5
+	innate_healing = TRUE
+
+	behavior_delegate_type = /datum/behavior_delegate/spider_guard
+	minimap_icon = "drone"
+
+/mob/living/carbon/xenomorph/spider_guard
+	caste_type = XENO_CASTE_SPIDER_GUARD
+	name = XENO_CASTE_SPIDER_GUARD
+	desc = "A giant arachnoid, furry and brown. It makes you shudder to look at it. This one has deep red eyes."
+	icon = 'icons/mob/xenos/spider_guard.dmi'
+	icon_size = 48
+	icon_state = "Spider Guard Running"
+	plasma_types = list(PLASMA_CHITIN)
+	tier = 1
+
+	speaking_noise = "click_talk"
+
+	slash_verb = "bite"
+	slashes_verb = "bites"
+	slash_sound = 'sound/weapons/bite.ogg'
+
+	acid_blood_damage = 0 /// it's a spider innit
+	acid_blood_spatter = FALSE /// We dont want that for these guys
+
+	icon_xeno = 'icons/mob/xenos/spider_guard.dmi'
+	icon_xenonid = 'icons/mob/xenos/spider_guard.dmi'
+
+	weed_food_icon = 'icons/mob/xenos/weeds.dmi'
+	weed_food_states = list("Hellhound_1","Hellhound_2","Hellhound_3")
+	weed_food_states_flipped = list("Hellhound_1","Hellhound_2","Hellhound_3")
+
+	base_actions = list(
+		/datum/action/xeno_action/onclick/xeno_resting,
+		/datum/action/xeno_action/onclick/plant_webs,
+	)
+	inherent_verbs = list(
+		/mob/living/carbon/xenomorph/proc/vent_crawl,
+	)
+	gib_chance = 1
+	claw_type = CLAW_TYPE_SHARP
+	pull_multiplier = 0.8
+
+/datum/behavior_delegate/spider_guard/melee_attack_additional_effects_target(mob/living/carbon/A)
+	if(prob(50))
+		A.apply_damage(5, TOX)
+		to_chat(A, SPAN_XENOHIGHDANGER("Your muscles sieze, as the [bound_xeno] bites into you with fangs that drip with venom!"))
+		A.sway_jitter(times = 2, steps = 1)
+		A.apply_effect(3, DAZE)
+		A.apply_effect(3, ROOT)
+
+/mob/living/carbon/xenomorph/spider_guard/death(cause, gibbed)
+	. = ..(cause, gibbed, "lets out a rattle as it collapses, legs siezing up.")
+	if(!.)
+		return
+
+/mob/living/carbon/xenomorph/spider_guard/handle_blood_splatter(splatter_dir)
+	new /obj/effect/temp_visual/dir_setting/bloodsplatter/yautjasplatter(loc, splatter_dir)
+
+	target_unconscious = FALSE
diff --git a/code/modules/mob/living/carbon/xenomorph/castes/Spider_Hunter.dm b/code/modules/mob/living/carbon/xenomorph/castes/Spider_Hunter.dm
new file mode 100644
index 00000000000..faf72c3db99
--- /dev/null
+++ b/code/modules/mob/living/carbon/xenomorph/castes/Spider_Hunter.dm
@@ -0,0 +1,182 @@
+/datum/caste_datum/spider_hunter //The sneaky ambusher
+	caste_type = XENO_CASTE_SPIDER_HUNTER //For when you want to hunt
+	tier = 1
+	melee_damage_lower = XENO_DAMAGE_TIER_3
+	melee_damage_upper = XENO_DAMAGE_TIER_4
+	melee_vehicle_damage = XENO_DAMAGE_TIER_2
+	max_health = XENO_HEALTH_SPIDER
+	plasma_gain = XENO_PLASMA_GAIN_TIER_8
+	plasma_max = XENO_PLASMA_TIER_10
+	xeno_explosion_resistance = XENO_EXPLOSIVE_ARMOR_TIER_1
+	armor_deflection = XENO_ARMOR_TIER_1
+	evasion = XENO_EVASION_MEDIUM
+	speed = XENO_SPEED_FASTSPIDER
+	fire_vulnerability_mult = FIRE_MULTIPLIER_DEADLY //v weak to fire
+
+	caste_desc = "An ambusher from the dark."
+	evolves_to = list() //XENO_CASTE_SPIDER_NURSE, XENO_CASTE_SPIDER_HUNTER
+	deevolves_to = list(XENO_CASTE_SPIDER_GUARD)
+
+	tackle_min = 3
+	tackle_max = 5
+	tackle_chance = 50
+	tacklestrength_min = 3
+	tacklestrength_max = 3
+
+	heal_resting = 3
+	heal_standing = 2
+	heal_knocked_out = 2
+	innate_healing = TRUE
+
+	behavior_delegate_type = /datum/behavior_delegate/spider_hunter
+	minimap_icon = "lurker"
+
+/mob/living/carbon/xenomorph/spider_hunter
+	caste_type = XENO_CASTE_SPIDER_HUNTER
+	name = XENO_CASTE_SPIDER_HUNTER
+	desc = "A giant arachnoid, furry and black. It makes you shudder to look at it. This one has glittering purple eyes."
+	icon = 'icons/mob/xenos/spider_hunter.dmi'
+	icon_size = 48
+	icon_state = "Spider Hunter Running"
+	plasma_types = list(PLASMA_CHITIN)
+	tier = 2
+
+	speaking_noise = "click_talk"
+
+	slash_verb = "bite"
+	slashes_verb = "bites"
+	slash_sound = 'sound/weapons/bite.ogg'
+
+	acid_blood_damage = 0 /// it's a spider innit
+	acid_blood_spatter = FALSE /// We dont want that for these guys
+
+	icon_xeno = 'icons/mob/xenos/spider_hunter.dmi'
+	icon_xenonid = 'icons/mob/xenos/spider_hunter.dmi'
+
+	weed_food_icon = 'icons/mob/xenos/weeds.dmi'
+	weed_food_states = list("Hellhound_1","Hellhound_2","Hellhound_3")
+	weed_food_states_flipped = list("Hellhound_1","Hellhound_2","Hellhound_3")
+
+	target_unconscious = FALSE
+
+	base_actions = list(
+		/datum/action/xeno_action/onclick/xeno_resting,
+		/datum/action/xeno_action/onclick/plant_webs,
+		/datum/action/xeno_action/activable/pounce/lurker,
+		/datum/action/xeno_action/onclick/lurker_invisibility,
+		/datum/action/xeno_action/onclick/lurker_assassinate,
+	)
+	inherent_verbs = list(
+		/mob/living/carbon/xenomorph/proc/vent_crawl,
+	)
+	gib_chance = 1
+	claw_type = CLAW_TYPE_SHARP
+	pull_multiplier = 0.5
+
+/mob/living/carbon/xenomorph/spider_hunter/death(cause, gibbed)
+	. = ..(cause, gibbed, "lets out a rattle as it collapses, legs siezing up.")
+	if(!.)
+		return
+
+/mob/living/carbon/xenomorph/spider_hunter/handle_blood_splatter(splatter_dir)
+	new /obj/effect/temp_visual/dir_setting/bloodsplatter/yautjasplatter(loc, splatter_dir)
+
+	forced_retarget_time = (2 SECONDS)
+
+/mob/living/carbon/xenomorph/spider_hunter/init_movement_handler()
+	return new /datum/xeno_ai_movement/linger/lurking(src)
+
+/datum/behavior_delegate/spider_hunter
+	name = "Base Spider Hunter Behavior Delegate"
+
+	// Config
+	var/invis_recharge_time = 30 SECONDS
+	var/invis_start_time = -1 // Special value for when we're not invisible
+	var/invis_duration = 20 SECONDS // so we can display how long the spider_hunter is invisible to it
+
+/datum/behavior_delegate/spider_hunter/melee_attack_additional_effects_target(mob/living/carbon/A)
+	if(prob(75))
+		A.apply_damage(15, TOX)
+		to_chat(A, SPAN_XENOHIGHDANGER("You feel woozy, as the [bound_xeno] bites into you with fangs that drip with venom!"))
+		A.sway_jitter(times = 5, steps = 3)
+		A.apply_effect(5, DAZE)
+		A.apply_effect(20, EYE_BLUR)
+
+	return
+
+/datum/behavior_delegate/spider_hunter/melee_attack_additional_effects_self()
+	..()
+
+	var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invis_action = get_action(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
+	if (lurker_invis_action)
+		lurker_invis_action.invisibility_off() // Full cooldown
+
+/datum/behavior_delegate/spider_hunter/proc/decloak_handler(mob/source)
+	SIGNAL_HANDLER
+	var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invis_action = get_action(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
+	if(istype(lurker_invis_action))
+		lurker_invis_action.invisibility_off(0.5) // Partial refund of remaining time
+
+/// Implementation for enabling invisibility.
+/datum/behavior_delegate/spider_hunter/proc/on_invisibility()
+	var/datum/action/xeno_action/activable/pounce/lurker/lurker_pounce_action = get_action(bound_xeno, /datum/action/xeno_action/activable/pounce/lurker)
+	if(lurker_pounce_action)
+		lurker_pounce_action.knockdown = TRUE // pounce knocks down
+		lurker_pounce_action.freeze_self = TRUE
+	ADD_TRAIT(bound_xeno, TRAIT_CLOAKED, TRAIT_SOURCE_ABILITY("cloak"))
+	RegisterSignal(bound_xeno, COMSIG_MOB_EFFECT_CLOAK_CANCEL, PROC_REF(decloak_handler))
+	bound_xeno.stealth = TRUE
+	invis_start_time = world.time
+
+/// Implementation for disabling invisibility.
+/datum/behavior_delegate/spider_hunter/proc/on_invisibility_off()
+	var/datum/action/xeno_action/activable/pounce/lurker/lurker_pounce_action = get_action(bound_xeno, /datum/action/xeno_action/activable/pounce/lurker)
+	if(lurker_pounce_action)
+		lurker_pounce_action.knockdown = FALSE // pounce no longer knocks down
+		lurker_pounce_action.freeze_self = FALSE
+	bound_xeno.stealth = FALSE
+	REMOVE_TRAIT(bound_xeno, TRAIT_CLOAKED, TRAIT_SOURCE_ABILITY("cloak"))
+	UnregisterSignal(bound_xeno, COMSIG_MOB_EFFECT_CLOAK_CANCEL)
+	invis_start_time = -1
+
+/datum/behavior_delegate/spider_hunter/append_to_stat()
+	. = list()
+
+	// Invisible
+	if(invis_start_time != -1)
+		var/time_left = (invis_duration-(world.time - invis_start_time)) / 10
+		. += "Invisibility Remaining: [time_left] second\s."
+		return
+
+	var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_action(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
+	if(!lurker_invisibility_action)
+		return
+
+	// Recharged
+	if(lurker_invisibility_action.cooldown_timer_id == TIMER_ID_NULL)
+		. += "Invisibility Recharge: Ready."
+		return
+
+	// Recharging
+	var/time_left = timeleft(lurker_invisibility_action.cooldown_timer_id) / 10
+	. += "Invisibility Recharge: [time_left] second\s."
+
+/datum/behavior_delegate/spider_hunter/on_collide(atom/movable/movable_atom)
+	. = ..()
+
+	if(!ishuman(movable_atom))
+		return
+
+	if(!bound_xeno || !bound_xeno.stealth)
+		return
+
+	var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_action(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
+	if(!lurker_invisibility_action)
+		return
+
+	var/mob/living/carbon/human/bumped_into = movable_atom
+	if(HAS_TRAIT(bumped_into, TRAIT_CLOAKED)) //ignore invisible scouts and preds
+		return
+
+	to_chat(bound_xeno, SPAN_XENOHIGHDANGER("We bumped into someone and lost our invisibility!"))
+	lurker_invisibility_action.invisibility_off(0.5) // partial refund of remaining time
diff --git a/code/modules/mob/living/carbon/xenomorph/castes/Spider_Nurse.dm b/code/modules/mob/living/carbon/xenomorph/castes/Spider_Nurse.dm
new file mode 100644
index 00000000000..c8d906e53d5
--- /dev/null
+++ b/code/modules/mob/living/carbon/xenomorph/castes/Spider_Nurse.dm
@@ -0,0 +1,126 @@
+/datum/caste_datum/spider_nurse //specialised spitter-esque enemy
+	caste_type = XENO_CASTE_SPIDER_NURSE //For when you want to send ranged spiders at the enemy
+	tier = 1
+	melee_damage_lower = XENO_DAMAGE_TIER_2
+	melee_damage_upper = XENO_DAMAGE_TIER_3
+	melee_vehicle_damage = XENO_DAMAGE_TIER_2
+	max_health = XENO_HEALTH_SPIDER
+	plasma_gain = XENO_PLASMA_GAIN_TIER_8
+	plasma_max = XENO_PLASMA_TIER_10
+	xeno_explosion_resistance = XENO_EXPLOSIVE_ARMOR_TIER_1
+	armor_deflection = XENO_ARMOR_TIER_1
+	evasion = XENO_EVASION_LOW
+	speed = XENO_SPEED_SPIDER
+	fire_vulnerability_mult = FIRE_MULTIPLIER_EXTREME //weak to fire
+
+	caste_desc = "A ranger of the cluster."
+	evolves_to = list() //XENO_CASTE_SPIDER_NURSE, XENO_CASTE_SPIDER_HUNTER
+	deevolves_to = list(XENO_CASTE_SPIDER_GUARD)
+
+	tackle_min = 2
+	tackle_max = 3
+	tackle_chance = 20
+
+	heal_resting = 4
+	heal_standing = 3
+	heal_knocked_out = 2
+	innate_healing = TRUE
+
+	behavior_delegate_type = /datum/behavior_delegate/spider_nurse
+	minimap_icon = "spitter"
+
+/mob/living/carbon/xenomorph/spider_nurse
+	caste_type = XENO_CASTE_SPIDER_NURSE
+	name = XENO_CASTE_SPIDER_NURSE
+	desc = "A giant arachnoid, furry in a pale hue. It makes you shudder to look at it. This one has shimmering green eyes."
+	icon = 'icons/mob/xenos/spider_nurse.dmi'
+	icon_size = 48
+	icon_state = "Spider Nurse Running"
+	plasma_types = list(PLASMA_CHITIN)
+	tier = 2
+
+	speaking_noise = "click_talk"
+
+	slash_verb = "bite"
+	slashes_verb = "bites"
+	slash_sound = 'sound/weapons/bite.ogg'
+
+	acid_blood_damage = 0 /// it's a spider innit
+	acid_blood_spatter = FALSE /// We dont want that for these guys
+
+	icon_xeno = 'icons/mob/xenos/spider_nurse.dmi'
+	icon_xenonid = 'icons/mob/xenos/spider_nurse.dmi'
+
+	weed_food_icon = 'icons/mob/xenos/weeds.dmi'
+	weed_food_states = list("Hellhound_1","Hellhound_2","Hellhound_3")
+	weed_food_states_flipped = list("Hellhound_1","Hellhound_2","Hellhound_3")
+	var/linger_range = 5
+	var/linger_deviation = 1
+
+	base_actions = list(
+		/datum/action/xeno_action/onclick/xeno_resting,
+		/datum/action/xeno_action/onclick/plant_webs,
+		/datum/action/xeno_action/activable/slowing_spit/ai, //first macro
+	)
+	inherent_verbs = list(
+		/mob/living/carbon/xenomorph/proc/vent_crawl,
+	)
+	gib_chance = 1
+	claw_type = CLAW_TYPE_SHARP
+	pull_multiplier = 1
+
+	target_unconscious = FALSE
+
+/datum/behavior_delegate/spider_nurse/melee_attack_additional_effects_target(mob/living/carbon/A)
+	if(prob(60))
+		A.apply_damage(10, TOX)
+		to_chat(A, SPAN_XENOHIGHDANGER("You feel woozy, as the [bound_xeno] bites into you with fangs that drip with venom!"))
+		A.sway_jitter(times = 3, steps = 2)
+		A.apply_effect(5, DAZE)
+		A.apply_effect(10, EYE_BLUR)
+
+/mob/living/carbon/xenomorph/spider_nurse/death(cause, gibbed)
+	. = ..(cause, gibbed, "lets out a rattle as it collapses, legs siezing up.")
+	if(!.)
+		return
+
+/mob/living/carbon/xenomorph/spider_nurse/handle_blood_splatter(splatter_dir)
+	new /obj/effect/temp_visual/dir_setting/bloodsplatter/yautjasplatter(loc, splatter_dir)
+
+/datum/behavior_delegate/spider_nurse
+	name = "Spider Nurse Behavior Delegate"
+
+/datum/action/xeno_action/activable/slowing_spit/ai
+	default_ai_action = TRUE
+	ai_prob_chance = 85
+	xeno_cooldown = 3 SECONDS
+
+
+/datum/action/xeno_action/activable/slowing_spit/ai/process_ai(mob/living/carbon/xenomorph/parent, delta_time)
+	/// Short-circuit. Will return the last thing checked or FALSE if it fails at any step.
+	/// We do not need to check for distance here as the tailstab itself will do that; that distance being 2.
+	return DT_PROB(ai_prob_chance, delta_time) && use_ability_async(parent.current_target) && (get_dist(parent, parent.current_target) <= 7) && !check_for_obstacles_projectile(parent, parent.current_target, GLOB.ammo_list[/datum/ammo/xeno/toxin])
+
+/proc/check_for_obstacles_projectile(mob/firer, mob/target, obj/projectile/P)
+	var/list/turf/path = get_line(firer, target, include_start_atom = FALSE)
+	if(!length(path) || get_dist(firer, target) > P.ammo.max_range)
+		return TRUE
+
+	var/blocked = FALSE
+	for(var/turf/T in path)
+		if(T.density && T.opacity)
+			blocked = TRUE
+			break
+
+		for(var/obj/O in T)
+			if(O.get_projectile_hit_boolean(P) && O.opacity)
+				blocked = TRUE
+				break
+
+	return blocked
+
+/mob/living/carbon/xenomorph/spider_nurse/init_movement_handler()
+	var/datum/xeno_ai_movement/linger/linger_movement = new(src)
+	linger_movement.linger_range = linger_range
+	linger_movement.linger_deviation = linger_deviation
+	return linger_movement
diff --git a/code/modules/mob/living/carbon/xenomorph/death.dm b/code/modules/mob/living/carbon/xenomorph/death.dm
index 5185468cc5a..2bf96c73ec4 100644
--- a/code/modules/mob/living/carbon/xenomorph/death.dm
+++ b/code/modules/mob/living/carbon/xenomorph/death.dm
@@ -1,7 +1,7 @@
 #define DELETE_TIME 1800
 
-/mob/living/carbon/xenomorph/death(cause, gibbed)
-	var/msg = "lets out a waning guttural screech, green blood bubbling from its maw."
+/mob/living/carbon/xenomorph/death(cause, gibbed, deathmessage)
+	var/msg = deathmessage ? deathmessage : "lets out a waning guttural screech, green blood bubbling from its maw."
 	. = ..(cause, gibbed, msg)
 	if(!.)
 		return //If they're already dead, it will return.
@@ -79,6 +79,10 @@
 			playsound(loc,'sound/voice/predalien_death.ogg', 25, TRUE)
 		else if(isfacehugger(src))
 			playsound(loc, 'sound/voice/alien_facehugger_dies.ogg', 25, TRUE)
+		else if (isspider(src))
+			playsound(loc, 'sound/voice/pred_click2.ogg', 25, TRUE)
+		else if (islizard(src))
+			playsound(loc, 'sound/effects/giant_lizard_death.ogg', 25, TRUE)
 		else
 			playsound(loc, prob(50) == 1 ? 'sound/voice/alien_death.ogg' : 'sound/voice/alien_death2.ogg', 25, 1)
 		var/area/A = get_area(src)
diff --git a/code/modules/mob/living/carbon/xenomorph/emote.dm b/code/modules/mob/living/carbon/xenomorph/emote.dm
index 572991016df..5b86c3a232f 100644
--- a/code/modules/mob/living/carbon/xenomorph/emote.dm
+++ b/code/modules/mob/living/carbon/xenomorph/emote.dm
@@ -4,6 +4,8 @@
 	keybind_category = CATEGORY_XENO_EMOTE
 	var/predalien_sound
 	var/larva_sound
+	var/spider_sound
+	var/lizard_sound
 
 /datum/emote/living/carbon/xeno/get_sound(mob/living/user)
 	. = ..()
@@ -14,11 +16,19 @@
 	if(islarva(user) && larva_sound)
 		. = larva_sound
 
+	if(isspider(user) && spider_sound)
+		. = spider_sound
+
+	if(islizard(user) && lizard_sound)
+		. = lizard_sound
+
 /datum/emote/living/carbon/xeno/growl
 	key = "growl"
 	message = "growls."
 	sound = "alien_growl"
 	predalien_sound = 'sound/voice/predalien_growl.ogg'
+	spider_sound = 'sound/voice/pred_click3.ogg'
+	lizard_sound = 'sound/effects/giant_lizard_growl1.ogg'
 	emote_type = EMOTE_AUDIBLE|EMOTE_VISIBLE
 
 /datum/emote/living/carbon/xeno/hiss
@@ -26,12 +36,16 @@
 	message = "hisses."
 	sound = "alien_hiss"
 	predalien_sound = 'sound/voice/predalien_hiss.ogg'
+	spider_sound = 'sound/voice/pred_click4.ogg'
+	lizard_sound = 'sound/effects/giant_lizard_hiss2.ogg'
 	emote_type = EMOTE_AUDIBLE|EMOTE_VISIBLE
 
 /datum/emote/living/carbon/xeno/needshelp
 	key = "needshelp"
 	message = "needs help!"
 	sound = "alien_help"
+	spider_sound = 'sound/voice/pred_click1.ogg'
+	lizard_sound = 'sound/effects/giant_lizard_hiss1.ogg'
 	emote_type = EMOTE_AUDIBLE|EMOTE_VISIBLE
 
 /datum/emote/living/carbon/xeno/roar
@@ -42,6 +56,8 @@
 	sound = "alien_roar"
 	predalien_sound = 'sound/voice/predalien_roar.ogg'
 	larva_sound = "alien_roar_larva"
+	spider_sound = 'sound/voice/pred_click5.ogg'
+	lizard_sound = 'sound/effects/giant_lizard_growl2.ogg'
 	emote_type = EMOTE_AUDIBLE|EMOTE_VISIBLE
 
 /datum/emote/living/carbon/xeno/tail
diff --git a/colonialmarines.dme b/colonialmarines.dme
index e2b70b1e956..307bb963026 100644
--- a/colonialmarines.dme
+++ b/colonialmarines.dme
@@ -2251,6 +2251,7 @@
 #include "code\modules\mob\living\carbon\xenomorph\ai\movement\facehugger.dm"
 #include "code\modules\mob\living\carbon\xenomorph\ai\movement\linger.dm"
 #include "code\modules\mob\living\carbon\xenomorph\ai\movement\lurking.dm"
+#include "code\modules\mob\living\carbon\xenomorph\ai\movement\spider.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Boiler.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Burrower.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Carrier.dm"
@@ -2259,6 +2260,7 @@
 #include "code\modules\mob\living\carbon\xenomorph\castes\Defender.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Drone.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Facehugger.dm"
+#include "code\modules\mob\living\carbon\xenomorph\castes\Giant_Lizard.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Hellhound.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Hivelord.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\king.dm"
@@ -2272,6 +2274,9 @@
 #include "code\modules\mob\living\carbon\xenomorph\castes\Runner.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Sentinel.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Soldier.dm"
+#include "code\modules\mob\living\carbon\xenomorph\castes\Spider_Guard.dm"
+#include "code\modules\mob\living\carbon\xenomorph\castes\Spider_Hunter.dm"
+#include "code\modules\mob\living\carbon\xenomorph\castes\Spider_Nurse.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Spitter.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Warrior.dm"
 #include "code\modules\mob\living\carbon\xenomorph\items\iff_tag.dm"
diff --git a/icons/mob/xenos/giant_lizard.dmi b/icons/mob/xenos/giant_lizard.dmi
new file mode 100644
index 00000000000..9e8bc87a5b7
Binary files /dev/null and b/icons/mob/xenos/giant_lizard.dmi differ
diff --git a/icons/mob/xenos/spider_guard.dmi b/icons/mob/xenos/spider_guard.dmi
new file mode 100644
index 00000000000..bf78cc05ce6
Binary files /dev/null and b/icons/mob/xenos/spider_guard.dmi differ
diff --git a/icons/mob/xenos/spider_hunter.dmi b/icons/mob/xenos/spider_hunter.dmi
new file mode 100644
index 00000000000..936cf043cea
Binary files /dev/null and b/icons/mob/xenos/spider_hunter.dmi differ
diff --git a/icons/mob/xenos/spider_nurse.dmi b/icons/mob/xenos/spider_nurse.dmi
new file mode 100644
index 00000000000..77fc2f32638
Binary files /dev/null and b/icons/mob/xenos/spider_nurse.dmi differ
