diff --git a/code/__DEFINES/mobs.dm b/code/__DEFINES/mobs.dm
index f6295f7a8d8..6ed639cc8d3 100644
--- a/code/__DEFINES/mobs.dm
+++ b/code/__DEFINES/mobs.dm
@@ -191,6 +191,7 @@
 #define GIVING (1<<5) // Is currently trying to give an item to someone
 #define NOBIOSCAN (1<<6)
 #define AI_CONTROLLED (1<<7)
+#define BODY_BURSTING (1<<8)
 
 //=================================================
 
diff --git a/code/__DEFINES/typecheck/xenos.dm b/code/__DEFINES/typecheck/xenos.dm
index 34b70ac92f4..6c844aa7f19 100644
--- a/code/__DEFINES/typecheck/xenos.dm
+++ b/code/__DEFINES/typecheck/xenos.dm
@@ -23,6 +23,8 @@
 #define isspitter(A) (istype(A, /mob/living/carbon/xenomorph/spitter))
 #define iswarrior(A) (istype(A, /mob/living/carbon/xenomorph/warrior))
 #define isburrower(A) (istype(A, /mob/living/carbon/xenomorph/burrower))
+#define islanky(A) (istype(A, /mob/living/carbon/xenomorph/lanky))
+// #define islankybrute(A) (istype(A, /mob/living/carbon/xenomorph/lankybrute))
 
 #define isxeno_builder(A) (isdrone(A) || ishivelord(A) || iscarrier(A) || isburrower(A) || isqueen(A))
 
diff --git a/code/__DEFINES/xeno.dm b/code/__DEFINES/xeno.dm
index b1a2545361e..4f17a2e0e51 100644
--- a/code/__DEFINES/xeno.dm
+++ b/code/__DEFINES/xeno.dm
@@ -107,6 +107,7 @@
 #define EGG_MAX_GROWTH_TIME 150
 
 #define FLAG_EMBRYO_PREDATOR 1 // Useful for passing down flags from an egg to the very moment an embryo is in someone
+#define FLAG_EMBRYO_HYBRID 2
 
 #define XENO_STARTING_CRYSTAL 100 //How much building resource the queen gets to start with
 
@@ -225,6 +226,7 @@
 #define XENO_ARMOR_TIER_4 35
 #define XENO_ARMOR_TIER_5 40
 #define XENO_ARMOR_TIER_7 50
+#define XENO_ARMOR_TIER_9 60
 
 // Explosive armor (think explosion resistance)
 #define XENO_NO_EXPLOSIVE_ARMOR 0
@@ -317,11 +319,13 @@
 #define XENO_SPEED_QUEEN 0.5
 #define XENO_SPEED_TIER_1 0.4
 #define XENO_SPEED_TIER_2 0.2
+#define XENO_SPEED_LANKY 0.1
 #define XENO_SPEED_TIER_3 0
 #define XENO_SPEED_TIER_4 -0.2
 #define XENO_SPEED_TIER_5 -0.4
 #define XENO_SPEED_TIER_6 -0.6
 #define XENO_SPEED_TIER_7 -0.8
+#define XENO_SPEED_RAPTOR -0.9
 #define XENO_SPEED_HELLHOUND -1
 #define XENO_SPEED_TIER_8 -1.4
 #define XENO_SPEED_TIER_9 -1.6
@@ -704,9 +708,12 @@
 #define XENO_CASTE_QUEEN  "Queen"
 #define XENO_CASTE_PREDALIEN  "Predalien"
 #define XENO_CASTE_HELLHOUND  "Hellhound"
+#define XENO_CASTE_BODYBURSTER  "Bodyburster"
+#define XENO_CASTE_LANKY  "Lanky"
+// #define XENO_CASTE_LANKYBRUTE  "Lanky Brute"
 #define XENO_SPECIAL_CASTES   list(XENO_CASTE_QUEEN, XENO_CASTE_PREDALIEN, XENO_CASTE_HELLHOUND)
 
-#define ALL_XENO_CASTES list(XENO_CASTE_LARVA, XENO_CASTE_PREDALIEN_LARVA, XENO_CASTE_FACEHUGGER, XENO_CASTE_LESSER_DRONE, XENO_CASTE_DRONE, XENO_CASTE_RUNNER, XENO_CASTE_SENTINEL, XENO_CASTE_DEFENDER, XENO_CASTE_BURROWER, XENO_CASTE_CARRIER, XENO_CASTE_HIVELORD, XENO_CASTE_LURKER, XENO_CASTE_WARRIOR, XENO_CASTE_SPITTER, XENO_CASTE_BOILER, XENO_CASTE_PRAETORIAN, XENO_CASTE_CRUSHER, XENO_CASTE_RAVAGER, XENO_CASTE_QUEEN, XENO_CASTE_PREDALIEN, XENO_CASTE_HELLHOUND, XENO_CASTE_SOLDIER, XENO_CASTE_KING)
+#define ALL_XENO_CASTES list(XENO_CASTE_LARVA, XENO_CASTE_PREDALIEN_LARVA, XENO_CASTE_FACEHUGGER, XENO_CASTE_LESSER_DRONE, XENO_CASTE_DRONE, XENO_CASTE_RUNNER, XENO_CASTE_SENTINEL, XENO_CASTE_DEFENDER, XENO_CASTE_BURROWER, XENO_CASTE_CARRIER, XENO_CASTE_HIVELORD, XENO_CASTE_LURKER, XENO_CASTE_WARRIOR, XENO_CASTE_SPITTER, XENO_CASTE_BOILER, XENO_CASTE_PRAETORIAN, XENO_CASTE_CRUSHER, XENO_CASTE_RAVAGER, XENO_CASTE_QUEEN, XENO_CASTE_PREDALIEN, XENO_CASTE_HELLHOUND, XENO_CASTE_SOLDIER, XENO_CASTE_KING, XENO_CASTE_BODYBURSTER, XENO_CASTE_LANKY)
 
 // Checks if two hives are allied to each other.
 // PARAMETERS:
diff --git a/code/game/jobs/role_authority.dm b/code/game/jobs/role_authority.dm
index 3cf4c46db49..5cbeb513e53 100644
--- a/code/game/jobs/role_authority.dm
+++ b/code/game/jobs/role_authority.dm
@@ -756,6 +756,12 @@ I hope it's easier to tell what the heck this proc is even doing, unlike previou
 			M = /mob/living/carbon/xenomorph/king
 		if(RUNNER_ACIDER)
 			M = /mob/living/carbon/xenomorph/runner/acider
+		if(XENO_CASTE_BODYBURSTER)
+			M = /mob/living/carbon/xenomorph/bodyburster
+		if(XENO_CASTE_LANKY)
+			M = /mob/living/carbon/xenomorph/lanky
+//		if(XENO_CASTE_LANKYBRUTE)
+//			M = /mob/living/carbon/xenomorph/lankybrute
 	return M
 
 
diff --git a/code/game/machinery/medical_pod/autodoc.dm b/code/game/machinery/medical_pod/autodoc.dm
index 35a38134564..690489fdaaa 100644
--- a/code/game/machinery/medical_pod/autodoc.dm
+++ b/code/game/machinery/medical_pod/autodoc.dm
@@ -529,11 +529,17 @@
 									S.limb_ref.implants -= I
 									H.embedded_items -= I
 									qdel(I)
-						var/obj/item/larva_ref = locate(/obj/item/alien_embryo) in H.contents
+						var/obj/item/alien_embryo/larva_ref = locate(/obj/item/alien_embryo) in H.contents
 						if(S.limb_ref.name == "chest" && larva_ref)
-							sleep(REMOVE_OBJECT_MAX_DURATION*surgery_mod)
-							H.contents -= larva_ref
-						qdel(larva_ref)
+							if(larva_ref.flags_embryo & FLAG_EMBRYO_HYBRID)
+								visible_message("[icon2html(src, viewers(src))] \The <b>[src]</b> croaks: Procedure has been deemed impossible.");
+								playsound(src.loc, 'sound/machines/buzz-two.ogg', 15, 1)
+								surgery_todo_list -= S
+								continue // Auto-doc cannot remove this embryo subtype
+							else
+								sleep(REMOVE_OBJECT_MAX_DURATION*surgery_mod)
+								H.contents -= larva_ref
+								qdel(larva_ref)
 						if(S.limb_ref.name == "chest" || S.limb_ref.name == "head")
 							close_encased(H,S.limb_ref)
 						if(!surgery) break
diff --git a/code/modules/admin/game_master/game_master.dm b/code/modules/admin/game_master/game_master.dm
index 4d3d24abafd..b1880f58f4d 100644
--- a/code/modules/admin/game_master/game_master.dm
+++ b/code/modules/admin/game_master/game_master.dm
@@ -42,8 +42,8 @@ GLOBAL_VAR_INIT(radio_communication_clarity, 100)
 	message_admins("[src] has [MODE_HAS_TOGGLEABLE_FLAG(MODE_NO_MAKE_BARRICADES) ? "disabled" : "enabled"] Barricade Creation.")
 
 // Spawn stuff
-#define DEFAULT_SPAWN_XENO_STRING XENO_CASTE_DRONE
-#define GAME_MASTER_AI_XENOS list(XENO_CASTE_DRONE, XENO_CASTE_SOLDIER, XENO_CASTE_RUNNER, RUNNER_ACIDER, XENO_CASTE_LURKER, XENO_CASTE_CRUSHER, XENO_CASTE_FACEHUGGER)
+#define DEFAULT_SPAWN_XENO_STRING XENO_CASTE_BODYBURSTER
+#define GAME_MASTER_AI_XENOS list(XENO_CASTE_LANKY, XENO_CASTE_BODYBURSTER, XENO_CASTE_DRONE, XENO_CASTE_SOLDIER, XENO_CASTE_RUNNER, RUNNER_ACIDER, XENO_CASTE_LURKER, XENO_CASTE_CRUSHER, XENO_CASTE_FACEHUGGER)
 #define DEFAULT_SPAWN_HIVE_STRING XENO_HIVE_NORMAL
 
 #define DEFAULT_XENO_AMOUNT_TO_SPAWN 1
diff --git a/code/modules/admin/game_master/game_master_submenu/infest.dm b/code/modules/admin/game_master/game_master_submenu/infest.dm
index 4d64e66e0bb..11e699dbd2b 100644
--- a/code/modules/admin/game_master/game_master_submenu/infest.dm
+++ b/code/modules/admin/game_master/game_master_submenu/infest.dm
@@ -54,15 +54,15 @@
 			return TRUE
 
 /datum/game_master_submenu/infest/proc/setup_embryo()
-	var/obj/item/alien_embryo/infesting_embryo
-	for(var/obj/item/alien_embryo/embryo in referenced_atom) //if this hive's embryo already exists, convert to larva and use it
+	var/obj/item/alien_embryo/bodyburster/infesting_embryo
+	for(var/obj/item/alien_embryo/bodyburster/embryo in referenced_atom) //if this hive's embryo already exists, convert to larva and use it
 		if(embryo.hivenumber == selected_hive)
 			infesting_embryo = embryo
 		else
 			qdel(embryo)
 
 	if(!infesting_embryo) //else, make a new one
-		infesting_embryo = new /obj/item/alien_embryo(referenced_atom)
+		infesting_embryo = new /obj/item/alien_embryo/bodyburster(referenced_atom)
 		infesting_embryo.hivenumber = selected_hive
 
 		var/mob/living/carbon/human/infested_host = referenced_atom
@@ -71,16 +71,16 @@
 	infesting_embryo.stage = embryo_stage
 
 /datum/game_master_submenu/infest/proc/remove_embryo()
-	for(var/obj/item/alien_embryo/embryo in referenced_atom)
+	for(var/obj/item/alien_embryo/bodyburster/embryo in referenced_atom)
 		qdel(embryo)
 
 /datum/game_master_submenu/infest/proc/force_burst()
-	var/mob/living/carbon/xenomorph/larva/infesting_larva = locate() in referenced_atom //if a larva already exists, use it
+	var/mob/living/carbon/xenomorph/bodyburster/infesting_larva = locate() in referenced_atom //if a larva already exists, use it
 	if(infesting_larva)
 		infesting_larva.chest_burst(referenced_atom)
 		return
 
-	for(var/obj/item/alien_embryo/embryo in referenced_atom) //else if this hive's embryo already exists, convert to larva and use it
+	for(var/obj/item/alien_embryo/bodyburster/embryo in referenced_atom) //else if this hive's embryo already exists, convert to larva and use it
 		if(embryo.hivenumber == selected_hive)
 			embryo.become_larva()
 			infesting_larva = locate() in referenced_atom
@@ -89,7 +89,7 @@
 		infesting_larva.chest_burst(referenced_atom)
 		return
 
-	infesting_larva = new /mob/living/carbon/xenomorph/larva(referenced_atom, null, selected_hive) //else, make a new larva
+	infesting_larva = new /mob/living/carbon/xenomorph/bodyburster(referenced_atom, null, selected_hive) //else, make a new larva
 	var/mob/living/carbon/human/infested_host = referenced_atom
 	infesting_larva.ckey = infested_host.ckey
 	infesting_larva.chest_burst(referenced_atom)
diff --git a/code/modules/admin/player_panel/actions/transform.dm b/code/modules/admin/player_panel/actions/transform.dm
index ca20829357d..217dea5f0de 100644
--- a/code/modules/admin/player_panel/actions/transform.dm
+++ b/code/modules/admin/player_panel/actions/transform.dm
@@ -99,6 +99,11 @@ GLOBAL_LIST_INIT(pp_transformables, list(
 	name = XENO_CASTE_CARRIER,
 	key = /mob/living/carbon/xenomorph/carrier,
 	color = "purple"
+	),
+	list(
+	name = XENO_CASTE_BODYBURSTER,
+	key = /mob/living/carbon/xenomorph/bodyburster,
+	color = "purple"
 	)
 	),
 
diff --git a/code/modules/mob/living/carbon/human/ai/brain/ai_brain_items.dm b/code/modules/mob/living/carbon/human/ai/brain/ai_brain_items.dm
index e4d88b3769e..c30328d6b5b 100644
--- a/code/modules/mob/living/carbon/human/ai/brain/ai_brain_items.dm
+++ b/code/modules/mob/living/carbon/human/ai/brain/ai_brain_items.dm
@@ -365,7 +365,9 @@
 /datum/human_ai_brain/proc/on_item_drop(datum/source, obj/item/dropped)
 	SIGNAL_HANDLER
 
-	if(dropped == primary_weapon)
+	if(dropped == null)
+		return
+	else if(dropped == primary_weapon)
 		if(!(gun_data.disposable && !primary_weapon.ai_can_use(tied_human, src)))
 			to_pickup |= dropped
 		set_primary_weapon(null)
diff --git a/code/modules/mob/living/carbon/xenomorph/Embryo.dm b/code/modules/mob/living/carbon/xenomorph/Embryo.dm
index 4a82c4fb0b8..5fc27d42231 100644
--- a/code/modules/mob/living/carbon/xenomorph/Embryo.dm
+++ b/code/modules/mob/living/carbon/xenomorph/Embryo.dm
@@ -47,6 +47,9 @@
 		qdel(src)
 		return FALSE
 
+	if((flags_embryo & FLAG_EMBRYO_HYBRID) && icon != 'icons/mob/xenos/bodyburster.dmi')
+		icon = 'icons/mob/xenos/bodyburster.dmi'
+
 	if(loc != affected_mob) //Our location is not the host
 		affected_mob.status_flags &= ~(XENO_HOST)
 		STOP_PROCESSING(SSobj, src)
@@ -60,15 +63,15 @@
 		if(ishuman(affected_mob))
 			var/mob/living/carbon/human/affected_human = affected_mob
 			if(world.time > affected_human.timeofdeath + affected_human.revive_grace_period) //Can't be defibbed.
-				var/mob/living/carbon/xenomorph/larva/larva_embryo = locate() in affected_mob
-				if(larva_embryo)
-					larva_embryo.chest_burst(affected_mob)
+				var/mob/living/carbon/xenomorph/embryo = locate() in affected_mob
+				if(embryo)
+					embryo.chest_burst(affected_mob)
 				qdel(src)
 				return FALSE
 		else
-			var/mob/living/carbon/xenomorph/larva/larva_embryo = locate() in affected_mob
-			if(larva_embryo)
-				larva_embryo.chest_burst(affected_mob)
+			var/mob/living/carbon/xenomorph/embryo = locate() in affected_mob
+			if(embryo)
+				embryo.chest_burst(affected_mob)
 			STOP_PROCESSING(SSobj, src)
 			return FALSE
 
@@ -91,6 +94,8 @@
 				counter += 0.11 * hive.larva_gestation_multiplier * delta_time
 		else if(HAS_TRAIT(affected_mob, TRAIT_NESTED)) //Hosts who are nested in resin nests provide an ideal setting, larva grows faster
 			counter += 1.5 * hive.larva_gestation_multiplier * delta_time //Currently twice as much, can be changed
+		else if(flags_embryo & FLAG_EMBRYO_HYBRID) //Hosts who have hybrid DNA in them grow faster
+			counter += 3 * hive.larva_gestation_multiplier * delta_time //very fast
 		else
 			if(stage < 5)
 				counter += 1 * hive.larva_gestation_multiplier * delta_time
@@ -105,28 +110,51 @@
 	switch(stage)
 		if(2)
 			if(prob(4))
-				if(!HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+				if(flags_embryo & FLAG_EMBRYO_HYBRID)
+					affected_mob.pain.apply_pain(PAIN_CHESTBURST_WEAK)
+					affected_mob.visible_message(SPAN_DANGER("[affected_mob] starts to shiver and tremble!"), \
+												SPAN_DANGER("You feel a slight ache in your bones, and your hairs stand on end."))
+					affected_mob.make_jittery(50)
+				else if(!HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
 					affected_mob.pain.apply_pain(PAIN_CHESTBURST_WEAK)
 					affected_mob.visible_message(SPAN_DANGER("[affected_mob] starts shaking uncontrollably!"), \
 												SPAN_DANGER("You feel something moving inside you! You start shaking uncontrollably!"))
 					affected_mob.apply_effect(1, PARALYZE)
 					affected_mob.make_jittery(105)
 					affected_mob.take_limb_damage(1)
-			if(prob(2))
-				var/message = SPAN_WARNING("[pick("Your chest hurts a little bit", "Your stomach hurts")].")
-				to_chat(affected_mob, message)
+
+			else if(prob(2))
+				if(flags_embryo & FLAG_EMBRYO_HYBRID)
+					var/message = SPAN_WARNING("[pick("Your skin prickles and twitches", "You feel an odd sensation in your bones")].")
+					to_chat(affected_mob, message)
+				else
+					var/message = SPAN_WARNING("[pick("Your chest hurts a little bit", "Your stomach hurts")].")
+					to_chat(affected_mob, message)
 		if(3)
 			if(prob(2))
-				var/message = SPAN_WARNING("[pick("Your throat feels sore", "Mucous runs down the back of your throat")].")
-				to_chat(affected_mob, message)
+				if(flags_embryo & FLAG_EMBRYO_HYBRID)
+					var/message = SPAN_WARNING("[pick("A trickle of sweat runs down your back", "Your muscles and bones ache")].")
+					to_chat(affected_mob, message)
+				else
+					var/message = SPAN_WARNING("[pick("Your throat feels sore", "Mucous runs down the back of your throat")].")
+					to_chat(affected_mob, message)
 			else if(prob(1))
 				to_chat(affected_mob, SPAN_WARNING("Your muscles ache."))
 				if(prob(20))
 					affected_mob.take_limb_damage(1)
 			else if(prob(2))
-				affected_mob.emote("[pick("sneeze", "cough")]")
+				if(flags_embryo & FLAG_EMBRYO_HYBRID)
+					affected_mob.emote("[pick("shiver", "twitch")]")
+				else
+					affected_mob.emote("[pick("sneeze", "cough")]")
 			if(prob(5))
-				if(!HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+				if(flags_embryo & FLAG_EMBRYO_HYBRID)
+					affected_mob.pain.apply_pain(PAIN_XENO_DRAG)
+					affected_mob.visible_message(SPAN_DANGER("[affected_mob] starts to shiver and tremble!"), \
+												SPAN_DANGER("You feel your muscles contort and tremble, and your bones twitch!"))
+					affected_mob.apply_effect(10, DAZE)
+					affected_mob.make_jittery(75)
+				else if(!HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
 					affected_mob.pain.apply_pain(PAIN_CHESTBURST_WEAK)
 					affected_mob.visible_message(SPAN_DANGER("\The [affected_mob] starts shaking uncontrollably!"), \
 												SPAN_DANGER("You feel something moving inside you! You start shaking uncontrollably!"))
@@ -142,7 +170,13 @@
 				if(prob(50))
 					affected_mob.emote("scream")
 			if(prob(6))
-				if(!HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+				if(flags_embryo & FLAG_EMBRYO_HYBRID)
+					affected_mob.pain.apply_pain(PAIN_XENO_GRAB)
+					affected_mob.visible_message(SPAN_DANGER("[affected_mob] starts to shiver and tremble!"), \
+												SPAN_DANGER("You feel your bones crack and shift, and your body begins to convulse!"))
+					affected_mob.apply_effect(20, DAZE)
+					affected_mob.make_jittery(100)
+				else if(!HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
 					affected_mob.pain.apply_pain(PAIN_CHESTBURST_WEAK)
 					affected_mob.visible_message(SPAN_DANGER("[affected_mob] starts shaking uncontrollably!"), \
 												SPAN_DANGER("You feel something moving inside you! You start shaking uncontrollably!"))
@@ -154,9 +188,9 @@
 		if(7) // Stage 6 is while we are trying to find a candidate in become_larva
 			larva_autoburst_countdown--
 			if(!larva_autoburst_countdown)
-				var/mob/living/carbon/xenomorph/larva/larva_embryo = locate() in affected_mob
-				if(larva_embryo)
-					larva_embryo.chest_burst(affected_mob)
+				var/mob/living/carbon/xenomorph/embryo = locate() in affected_mob
+				if(embryo)
+					embryo.chest_burst(affected_mob)
 
 ///We look for a candidate. If found, we spawn the candidate as a larva
 ///Order of priority is bursted individual (if xeno is enabled), then player hugger, then random candidate, and then it's up for grabs and spawns braindead
@@ -239,14 +273,16 @@
 							break
 
 	// Spawn the larva
-	var/mob/living/carbon/xenomorph/larva/new_xeno
+	var/mob/living/carbon/xenomorph/new_xeno
 
 	if(isyautja(affected_mob) || (flags_embryo & FLAG_EMBRYO_PREDATOR))
 		new_xeno = new /mob/living/carbon/xenomorph/larva/predalien(affected_mob)
 		yautja_announcement(SPAN_YAUTJABOLDBIG("WARNING!\n\nAn abomination has been detected at [get_area_name(new_xeno)]. It is a stain upon our purity and is unfit for life. Exterminate it immediately.\n\nHeavy Armory unlocked."))
 		SEND_GLOBAL_SIGNAL(COMSIG_GLOB_YAUTJA_ARMORY_OPENED)
+	if(flags_embryo & FLAG_EMBRYO_HYBRID)
+		new_xeno = new /mob/living/carbon/xenomorph/bodyburster(affected_mob)
 	else
-		new_xeno = new(affected_mob)
+		new_xeno = new /mob/living/carbon/xenomorph/larva(affected_mob)
 
 	if(hive)
 		hive.add_xeno(new_xeno)
@@ -282,7 +318,10 @@
 
 	stage = 7 // Begin the autoburst countdown
 
-/mob/living/carbon/xenomorph/larva/proc/cause_unbearable_pain(mob/living/carbon/victim)
+/mob/living/carbon/xenomorph/proc/cause_unbearable_pain(mob/living/carbon/victim)
+	return
+
+/mob/living/carbon/xenomorph/larva/cause_unbearable_pain(mob/living/carbon/victim)
 	if(loc != victim)
 		return
 	victim.emote("scream")
@@ -292,7 +331,10 @@
 	to_chat(victim, message)
 	addtimer(CALLBACK(src, PROC_REF(cause_unbearable_pain), victim), rand(1, 3) SECONDS, TIMER_UNIQUE|TIMER_NO_HASH_WAIT)
 
-/mob/living/carbon/xenomorph/larva/proc/chest_burst(mob/living/carbon/victim)
+/mob/living/carbon/xenomorph/proc/chest_burst(mob/living/carbon/victim)
+	return
+
+/mob/living/carbon/xenomorph/larva/chest_burst(mob/living/carbon/victim)
 	set waitfor = 0
 	if(victim.chestburst || loc != victim)
 		return
diff --git a/code/modules/mob/living/carbon/xenomorph/castes/Bodyburster.dm b/code/modules/mob/living/carbon/xenomorph/castes/Bodyburster.dm
new file mode 100644
index 00000000000..2434eb841ef
--- /dev/null
+++ b/code/modules/mob/living/carbon/xenomorph/castes/Bodyburster.dm
@@ -0,0 +1,354 @@
+/datum/caste_datum/bodyburster
+	caste_type = XENO_CASTE_BODYBURSTER
+	tier = 2
+
+	melee_damage_lower = XENO_DAMAGE_TIER_4
+	melee_damage_upper = XENO_DAMAGE_TIER_5
+	melee_vehicle_damage = XENO_DAMAGE_TIER_4
+	max_health = XENO_HEALTH_TIER_2
+	plasma_gain = XENO_PLASMA_GAIN_TIER_8
+	plasma_max = XENO_PLASMA_TIER_10
+	xeno_explosion_resistance = XENO_EXPLOSIVE_ARMOR_TIER_4
+	armor_deflection = XENO_ARMOR_TIER_3
+	evasion = XENO_EVASION_MEDIUM
+	speed = XENO_SPEED_RAPTOR
+
+	evolves_to = list()
+	deevolves_to = list()
+	caste_desc = "A powerful new xenogenetic abberation."
+	can_vent_crawl = 0
+
+	tackle_min = 2
+	tackle_max = 4
+
+	minimap_icon = "warrior"
+
+	heal_resting = 1.4
+
+/mob/living/carbon/xenomorph/bodyburster
+	caste_type = XENO_CASTE_BODYBURSTER
+	name = XENO_CASTE_BODYBURSTER
+	desc = "A horrific abberation of the usual alien, with an armored carapace. The human skull is almost clearly visible within its head, and bits of skin and gore still cling to crevices in its body."
+	icon = 'icons/mob/xenos/bodyburster.dmi'
+	icon_size = 48
+	icon_state = "Normal Bodyburster Walking"
+	plasma_types = list(PLASMA_CATECHOLAMINE)
+	pixel_x = -8
+	old_x = -8
+	tier = 2
+	pull_speed = 2 // about what it was before, slightly faster
+	organ_value = 2000
+
+	claw_type = CLAW_TYPE_SHARP
+	gib_chance = 100
+
+	icon_xeno = 'icons/mob/xenos/bodyburster.dmi'
+	icon_xenonid = 'icons/mob/xenonids/warrior.dmi'
+	acid_blood_damage = 45 /// yowch!
+	acid_blood_spatter = TRUE /// They should be dangerous in close quarters
+
+	weed_food_icon = 'icons/mob/xenos/weeds_64x64.dmi'
+	weed_food_states = list("Warrior_1","Warrior_2","Warrior_3")
+	weed_food_states_flipped = list("Warrior_1","Warrior_2","Warrior_3")
+
+/*
+ * ==========================================================================|
+ * 							BASE DEFINES AND PROCS
+ * --------------------------------------------------------------------------|
+ * ==========================================================================|
+*/
+
+/// The aggression the alien starts with by default.
+#define AGGRESSION_MINIMUM 0
+/// The maximum aggression it is possible to accumulate.
+#define AGGRESSION_MAXIMUM 100
+/// Aggression required to go full-on lethals instead of capture.
+#define AGGRESSION_ENABLE_LETHAL 40
+/// The cap on aggression increase when something significant happens, like taking damage.
+#define AGGRESSION_INCREMENT_CAP 10
+
+/// Threshold for lunging at a target. Low as the alien wants to do this early.
+#define AGGRESSION_LUNGE 10
+/// Threshold for throwing humans around.
+#define AGGRESSION_FLING 70
+/// Threshold for stabbing them with the tail.
+#define AGGRESSION_TAIL_STAB 50
+/// Threshold for headbiting/instakilling a downed human.
+#define AGGRESSION_HEADBITE 90
+
+	/// AI behavior variables should probably go into their own datum. I thought about placing this into movement handler, but these govern general behavior.
+	var/aggression_cur = AGGRESSION_MINIMUM ///How angry this alien is. Higher aggression means more lethal options are open.
+	var/aggression_min = AGGRESSION_MINIMUM ///Aggression cannot fall below this value.
+
+/mob/living/carbon/xenomorph/bodyburster/Initialize()
+	base_actions = list(
+		/datum/action/xeno_action/onclick/xeno_resting,
+		/datum/action/xeno_action/onclick/regurgitate,
+		/datum/action/xeno_action/watch_xeno,
+		/datum/action/xeno_action/activable/tail_stab/bodyburster,
+		/datum/action/xeno_action/activable/fling/bodyburster,
+		/datum/action/xeno_action/activable/lunge/bodyburster,
+		/datum/action/xeno_action/activable/headbite/bodyburster,
+		/datum/action/xeno_action/onclick/tacmap,
+	)
+	inherent_verbs = list(
+		/mob/living/carbon/xenomorph/proc/vent_crawl,
+	)
+
+	. = ..()
+
+/mob/living/carbon/xenomorph/bodyburster/apply_damage(damage = 0, damagetype = BRUTE, def_zone = null, used_weapon = null, sharp = 0, edge = 0, force = FALSE)
+	. = ..() //It will take the previous arguments.
+
+	if(.)
+		/// Fire gives the largest aggression increase.
+		aggression_cur = min(AGGRESSION_MAXIMUM, aggression_cur + min(AGGRESSION_INCREMENT_CAP, damage * (damagetype == BURN ? 0.25 : 0.17) ) ) /// Multiplying is faster than dividing. Go figure.
+
+/*
+ * ==========================================================================|
+ * 						MOVEMENT LOOPS AND BEHAVIOR LOGIC
+ * --------------------------------------------------------------------------|
+ * ==========================================================================|
+*/
+
+/mob/living/carbon/xenomorph/bodyburster/process_ai(delta_time)
+	var/mob/living/pulling_target = pulling /// Let's see if the alien is pulling anyone.
+	var/mob/living/potential_target = current_target
+
+	if(istype(pulling_target)) /// Our bodyburster is pulling someone.
+		if(get_active_hand()) swap_hand() /// Swap hand to either tackle or harm.
+		ai_active_intent = INTENT_DISARM /// If we are pulling someone and are not too aggressive, switch to disarm.
+		if(prob(5)) emote("tail")
+
+	else if(istype(potential_target)) /// We have a target. We'll do more thorough checking in the main loop, for now we only need to know if they are being pulled by a hostile or friendly xeno.
+		var/mob/living/carbon/xenomorph/other_xenomorph = potential_target.pulledby /// Are they being pulled by an alien?
+		/// Need to make sure the alien dragging is friendly to us. If it is not friendly, or not a xeno, our alien will try to grab back.
+		ai_active_intent = (istype(other_xenomorph) && IS_SAME_HIVENUMBER(src, other_xenomorph)) ? INTENT_DISARM : INTENT_GRAB
+
+	/// I had it set up for slightly faster assignment, but this is easier to read.
+	ai_active_intent = (aggression_cur >= AGGRESSION_ENABLE_LETHAL) ? INTENT_HARM : ai_active_intent /// Override harm or continue with the previous intent.
+
+	return ..()
+
+/datum/xeno_ai_movement/assault
+	do_climb_structures = TRUE /// The alien will climb structures, if able.
+
+/mob/living/carbon/xenomorph/bodyburster/init_movement_handler()
+	return new /datum/xeno_ai_movement/assault(src)
+
+/mob/living/carbon/xenomorph/bodyburster/ai_move_idle(delta_time)
+	if(!ai_movement_handler)
+		CRASH("No valid movement handler for [src]!")
+
+	var/mob/living/pulling_target = pulling
+	return (istype(pulling_target) && length(GLOB.ai_hives)) ? ai_movement_handler.ai_move_hive(delta_time) : ai_movement_handler.ai_move_idle(delta_time)
+
+/mob/living/carbon/xenomorph/bodyburster/ai_move_target(delta_time)
+	if(!ai_movement_handler)
+		CRASH("No valid movement handler for [src]!")
+
+	var/mob/living/pulling_target = pulling
+	return (istype(pulling_target) && length(GLOB.ai_hives)) ? ai_movement_handler.ai_move_hive(delta_time) : ai_movement_handler.ai_move_target(delta_time)
+
+/datum/xeno_ai_movement/assault/ai_move_target(delta_time)
+	var/mob/living/carbon/xenomorph/bodyburster/current_parent = parent
+
+	/// Moving toward target grows aggression.
+	current_parent.aggression_cur = min(AGGRESSION_MAXIMUM, current_parent.aggression_cur + 0.1) ///Fractions for a more gradual aggression gain/loss.
+	return ..()
+
+/datum/xeno_ai_movement/assault/ai_move_idle(delta_time)
+	var/mob/living/carbon/xenomorph/bodyburster/current_parent = parent
+
+	/// Being idle drains aggression.
+	current_parent.aggression_cur = max(current_parent.aggression_min, current_parent.aggression_cur - 0.1)
+	return ..()
+
+/*
+ * ==========================================================================|
+ * 							SPECIAL ACTIONS
+ * --------------------------------------------------------------------------|
+ * ==========================================================================|
+*/
+
+/mob/living/carbon/xenomorph/bodyburster/check_additional_ai_activation(activation_threshold)
+	return (aggression_cur >= activation_threshold)
+
+/datum/action/xeno_action/activable/tail_stab/bodyburster /// Specific to bodybursters, so that not all xenos get it.
+	default_ai_action = TRUE
+	ai_prob_chance = 50 //So they are not spamming it quite as often.
+	charge_time = null /// AI bodybursters can just use this whenever instead of having to charge it up.
+	xeno_cooldown = 30 SECONDS
+
+/datum/action/xeno_action/activable/tail_stab/process_ai(mob/living/carbon/xenomorph/parent, delta_time)
+	/// Short-circuit. Will return the last thing checked or FALSE if it fails at any step.
+	/// We do not need to check for distance here as the tailstab itself will do that; that distance being 2.
+	return parent.check_additional_ai_activation(AGGRESSION_TAIL_STAB) && DT_PROB(ai_prob_chance, delta_time) && use_ability_async(parent.current_target)
+
+/datum/action/xeno_action/activable/fling/bodyburster
+	default_ai_action = TRUE
+	ai_prob_chance = 35
+	xeno_cooldown = 15 SECONDS
+
+/datum/action/xeno_action/activable/fling/process_ai(mob/living/carbon/xenomorph/parent, delta_time)
+	/// We have a home turf to fling to.
+	if(parent.check_additional_ai_activation(AGGRESSION_FLING) && DT_PROB(ai_prob_chance, delta_time))
+		parent.dir = parent.ai_movement_handler.home_turf ? get_dir(parent, parent.ai_movement_handler.home_turf) : pick(NORTH, SOUTH, EAST, WEST) /// Pick at random if there is no valid direction.
+		use_ability_async(parent.current_target)
+
+/datum/action/xeno_action/activable/lunge/bodyburster
+	default_ai_action = TRUE
+	grab_range = 3
+	ai_prob_chance = 90 // Want to do this often, as it's their way of saying hello.
+	xeno_cooldown = 15 SECONDS
+
+/datum/action/xeno_action/activable/lunge/process_ai(mob/living/carbon/xenomorph/parent, delta_time)
+	/// Want to make sure no obstacles are in the way so that the alien is not lunging for no reason, or bonking into barricades like an idiot.
+	/// Maybe in the future the actual lunge can be stripped down for the AI only?
+	if( parent.check_additional_ai_activation(AGGRESSION_LUNGE) && DT_PROB(ai_prob_chance, delta_time) && get_dist(parent, parent.current_target) == grab_range )
+		/// get_step_to() should return the turf nearest the target if successful, with no obstacles to block movement there with the lunge.
+		var/turf/T = get_step_to(parent, parent.current_target)
+		return T?.AdjacentQuick(parent.current_target.loc) && use_ability_async(parent.current_target)
+
+/// Override for bodyburster lunges. Similar to the woyer grab, but this one only inflicts pain without stuns and has a shorter range.
+/mob/living/carbon/xenomorph/bodyburster/start_pulling(atom/movable/AM, lunge)
+	if (!check_state() || agility || !isliving(AM)) return FALSE
+
+	var/mob/living/L = AM
+	var/should_neckgrab = !can_not_harm(L) && lunge
+
+	if(!QDELETED(L) && !QDELETED(L.pulledby) && L != src ) //override pull of other mobs
+		visible_message(SPAN_WARNING("[src] has broken [L.pulledby]'s grip on [L]!"), null, null, 5)
+		L.pulledby.stop_pulling()
+
+	. = ..(L, lunge, should_neckgrab)
+
+	if(.)
+		var/pain_to_cause = PAIN_XENO_DRAG /// Basic amount of pain caused with each grab.
+		if(should_neckgrab && L.mob_size < MOB_SIZE_BIG)
+			pain_to_cause += PAIN_XENO_GRAB /// Neck grabs cause even more pain.
+			L.pulledby = src
+			visible_message(SPAN_XENOWARNING("\The [src] grabs [L] by the throat!"), \
+			SPAN_XENOWARNING("You grab [L] by the throat!"))
+
+		L.pain.apply_pain(pain_to_cause)
+		grab_level = GRAB_XENO /// Alien-specific grab level, with its own logic for escaping. AI only for the moment. See /mob/living/resist_grab()
+		if(prob(10)) emote("growl")
+		///	The actual pain processing for humans is handled in: /mob/living/carbon/proc/handle_grabbed() Other mobs don't process the effects of the grab, like other xenomorphs.
+
+/mob/living/carbon/xenomorph/bodyburster/stop_pulling(bumped_movement = FALSE)
+	//Let's see if we can ignore this. If our direction is the same as where the mob went, we likely bumped into them. So we lasso them back.
+	if(bumped_movement && grab_level == GRAB_XENO && get_dir(src, pulling) == dir) /// Only on xeno grabs, so if they are on GRAB_PASSIVE, this doesn't trigger.
+		pulling.loc = get_step_towards(src, pulling) // GET OVER HERE!
+	else return ..()
+
+/datum/action/xeno_action/activable/headbite/bodyburster
+	default_ai_action = TRUE
+	ai_prob_chance = 95 // Absolutely DEAD if the alien is angry enough.
+	xeno_cooldown = 30 SECONDS // Don't want to chain these, as unlikely as that could be.
+
+/datum/action/xeno_action/activable/headbite/bodyburster/process_ai(mob/living/carbon/xenomorph/parent, delta_time)
+	return parent.check_additional_ai_activation(AGGRESSION_HEADBITE) && DT_PROB(ai_prob_chance, delta_time) && use_ability_async(parent.current_target)
+
+/mob/living/carbon/xenomorph/bodyburster/initialize_pass_flags(datum/pass_flags_container/pass_flags_container)
+	..()
+	if (pass_flags_container)
+		pass_flags_container.flags_pass |= PASS_FLAGS_CRAWLER
+
+
+/mob/living/carbon/xenomorph/bodyburster/cause_unbearable_pain(mob/living/carbon/victim)
+	if(loc != victim)
+		return
+	if(prob(15))
+		victim.pain.apply_pain(PAIN_XENO_DRAG)
+	victim.visible_message(SPAN_DANGER("\The [victim] is deathly pale, sweat beading across their skin."),
+						SPAN_DANGER(pick("YOU FEEL DEATHLY COLD!", "YOU ARE BURNING UP!", "MAKE IT STOP!", "SOMETHING IS WRONG WITH YOUR BODY!", "YOUR MUSCLES SHIVER AND TWITCH UNCONTROLLABLY!", "YOUR BONES FEEL LIKE THEY ARE BREAKING!", "YOU CAN'T BREATHE!", "EVERYTHING HURTS!")))
+	addtimer(CALLBACK(src, PROC_REF(cause_unbearable_pain), victim), rand(3, 5) SECONDS, TIMER_UNIQUE|TIMER_NO_HASH_WAIT)
+
+/mob/living/carbon/xenomorph/bodyburster/chest_burst(mob/living/carbon/victim)
+	set waitfor = 0
+	if(victim.chestburst || loc != victim)
+		return
+	victim.mob_flags |= BODY_BURSTING
+	victim.chestburst = TRUE
+	to_chat(src, SPAN_DANGER("We start emerging from [victim]'s body!"))
+	if(!HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+		victim.apply_effect(20, SUPERSLOW)
+	victim.visible_message(SPAN_DANGER("\The [victim] starts to shake uncontrollably, their flesh giving way to a new form!"),
+						SPAN_DANGER("YOU FEEL YOUR BODY CHURN AND SHIFT!"))
+	victim.make_jittery(200)
+	sleep(10)
+	if(!victim || !victim.loc)
+		return//host could've been deleted, or we could've been removed from host.
+	if(loc != victim)
+		victim.chestburst = 0
+		return
+	if(ishuman(victim) || isyautja(victim))
+		victim.emote("burstscream")
+	sleep(25) //Sound delay
+	victim.update_burst()
+	sleep(10) //Sprite delay
+	playsound(victim, 'sound/effects/gibbed.ogg', 50)
+	if(!victim || !victim.loc)
+		return
+	if(loc != victim)
+		victim.chestburst = 0 //if a doc removes the larva during the sleep(10), we must remove the 'bursting' overlay on the human
+		victim.update_burst()
+		return
+
+	var/burstcount = 0
+
+	victim.spawn_gibs()
+
+	for(var/mob/living/carbon/xenomorph/bodyburster/burster_embryo in victim)
+		var/datum/hive_status/hive = GLOB.hive_datum[burster_embryo.hivenumber]
+		burster_embryo.forceMove(get_turf(victim)) //moved to the turf directly so we don't get stuck inside a cryopod or another mob container.
+		var/list/splitname = splittext(victim.real_name, " ")
+		burster_embryo.name = "[splitname[1]]omorph ([burster_embryo.nicknumber])"
+		burster_embryo.real_name = burster_embryo.name
+		playsound(burster_embryo, pick('sound/voice/alien_roar1.ogg', 'sound/voice/alien_roar2.ogg', 'sound/voice/alien_roar3.ogg', 'sound/voice/alien_roar4.ogg', 'sound/voice/alien_roar5.ogg', 'sound/voice/alien_roar6.ogg'), 30)
+
+		if(burster_embryo.client)
+			burster_embryo.set_lighting_alpha_from_prefs(burster_embryo.client)
+
+		burster_embryo.attack_log += "\[[time_stamp()]\]<font color='red'> bodybursted from [key_name(victim)] in [get_area_name(burster_embryo)] at X[victim.x], Y[victim.y], Z[victim.z]</font>"
+		victim.attack_log += "\[[time_stamp()]\]<font color='orange'> Was bodybursted in [get_area_name(burster_embryo)] at X[victim.x], Y[victim.y], Z[victim.z]. The larva was [key_name(burster_embryo)].</font>"
+
+		if(burstcount)
+			step(burster_embryo, pick(GLOB.cardinals))
+
+		if(GLOB.round_statistics && (ishuman(victim)) && (SSticker.current_state == GAME_STATE_PLAYING) && (ROUND_TIME > 1 MINUTES))
+			GLOB.round_statistics.total_larva_burst++
+		GLOB.larva_burst_by_hive[hive] = (GLOB.larva_burst_by_hive[hive] || 0) + 1
+		burstcount++
+
+		if(!victim.first_xeno)
+			if(hive.hive_orders)
+				to_chat(burster_embryo, SPAN_XENOHIGHDANGER("The unknown force guides us to our new form..."))
+				to_chat(burster_embryo, SPAN_XENOHIGHDANGER("\"[hive.hive_orders]\""))
+			log_attack("[key_name(victim)] bodyburst in [get_area_name(burster_embryo)] at X[victim.x], Y[victim.y], Z[victim.z]. The bodyburster was [key_name(burster_embryo)].") //this is so that admins are not spammed with los logs
+
+	for(var/list/obj/limb/limbs in victim)
+		if(istype(limbs, /obj/limb/chest) || istype(limbs, /obj/limb/groin))
+			continue
+		qdel(limbs)
+
+	for(var/obj/item/alien_embryo/AE in victim)
+		qdel(AE)
+
+	var/datum/cause_data/cause = create_cause_data("bodybursting", src)
+	victim.gib(cause)
+
+/obj/item/alien_embryo/bodyburster
+	icon = 'icons/mob/xenos/bodyburster.dmi'
+	flags_embryo = FLAG_EMBRYO_HYBRID
+
+#undef AGGRESSION_MINIMUM
+#undef AGGRESSION_MAXIMUM
+#undef AGGRESSION_INCREMENT_CAP
+#undef AGGRESSION_ENABLE_LETHAL
+
+#undef AGGRESSION_LUNGE
+#undef AGGRESSION_FLING
+#undef AGGRESSION_TAIL_STAB
+#undef AGGRESSION_HEADBITE
diff --git a/code/modules/mob/living/carbon/xenomorph/castes/Lanky.dm b/code/modules/mob/living/carbon/xenomorph/castes/Lanky.dm
new file mode 100644
index 00000000000..aaa6f40ae9a
--- /dev/null
+++ b/code/modules/mob/living/carbon/xenomorph/castes/Lanky.dm
@@ -0,0 +1,100 @@
+/datum/caste_datum/lanky
+	caste_type = XENO_CASTE_LANKY
+	tier = 3
+
+	melee_damage_lower = XENO_DAMAGE_TIER_6
+	melee_damage_upper = XENO_DAMAGE_TIER_6
+	melee_vehicle_damage = XENO_DAMAGE_TIER_5 * 10
+	max_health = XENO_HEALTH_IMMORTAL
+	plasma_gain = XENO_PLASMA_GAIN_TIER_7
+	plasma_max = XENO_PLASMA_TIER_8
+	xeno_explosion_resistance = XENO_EXPLOSIVE_ARMOR_TIER_10
+	armor_deflection = XENO_ARMOR_TIER_9
+	evasion = XENO_EVASION_NONE
+	speed = XENO_SPEED_LANKY
+	heal_standing = 0.66
+
+	evolution_allowed = TRUE
+	evolves_to = list()
+	deevolves_to = list()
+	caste_desc = "The typical bipedal Lankies when they seed worlds."
+
+	tackle_min = 2
+	tackle_max = 6
+	tackle_chance = 25
+
+	behavior_delegate_type = /datum/behavior_delegate/lanky
+	minimap_icon = "runner"
+
+/mob/living/carbon/xenomorph/lanky
+	caste_type = XENO_CASTE_LANKY
+	name = XENO_CASTE_LANKY
+	desc = "A tall, bipedal alien with strange growths and a thick exoskeleton."
+	icon = 'icons/mob/xenos/lanky.dmi'
+	icon_size = 64
+	icon_state = "Normal Lanky Walking"
+	plasma_types = list(PLASMA_CHITIN)
+	pixel_x = -16
+	old_x = -16
+	mob_size = MOB_SIZE_BIG
+	drag_delay = 6 //pulling a big dead xeno is hard
+	tier = 3
+	organ_value = 3000
+
+	small_explosives_stun = FALSE
+
+	acid_blood_damage = 0 /// it's a lanky innit
+	acid_blood_spatter = FALSE /// We dont want that for these guys
+
+	mob_size = MOB_SIZE_IMMOBILE
+
+	rebounds = FALSE // no more fucking pinball crooshers
+	base_actions = list(
+		/datum/action/xeno_action/onclick/xeno_resting,
+		/datum/action/xeno_action/watch_xeno,
+		/datum/action/xeno_action/onclick/crusher_stomp,
+		/datum/action/xeno_action/onclick/tacmap,
+	)
+
+	gib_chance = 0
+	claw_type = CLAW_TYPE_VERY_SHARP
+
+	icon_xeno = 'icons/mob/xenos/lanky.dmi'
+	icon_xenonid = 'icons/mob/xenonids/praetorian.dmi'
+
+	weed_food_icon = 'icons/mob/xenos/weeds_64x64.dmi'
+	weed_food_states = list("Praetorian_1","Praetorian_2","Praetorian_3")
+	weed_food_states_flipped = list("Praetorian_1","Praetorian_2","Praetorian_3")
+
+	ai_range = 24
+	forced_retarget_time = (3 SECONDS)
+
+/mob/living/carbon/xenomorph/lanky/Initialize(mapload, mob/living/carbon/xenomorph/oldXeno, h_number, ai_hard_off = FALSE)
+	. = ..()
+	AddComponent(/datum/component/footstep, 2, 50, 15, 1, "metalbang")
+
+	playsound(src, 'sound/voice/xeno_praetorian_screech.ogg', 100, TRUE, 30, falloff = 5)
+	if(!get_turf(src)) //autowiki compat, spawns in nullspace
+		return
+	for(var/mob/current_mob as anything in get_mobs_in_z_level_range(get_turf(src), 30) - src)
+		var/relative_dir = get_dir(current_mob, src)
+		var/final_dir = dir2text(relative_dir)
+		to_chat(current_mob, SPAN_HIGHDANGER("You hear a terrible screech coming from [final_dir ? "the [final_dir]" : "nearby"] as the ground shakes!"))
+		for(var/turf/closed/wall/wall in orange(1, src))
+			wall.ex_act(1000 * 1.5)
+
+
+/mob/living/carbon/xenomorph/lanky/death(cause, gibbed)
+	. = ..()
+	playsound(src, 'sound/voice/alien_crusher_death.ogg', 50, FALSE, 15)
+
+/datum/behavior_delegate/lanky/melee_attack_additional_effects_target(mob/living/carbon/A)
+	if(prob(50))
+		A.apply_damage(5, OXY)
+		to_chat(A, SPAN_XENOHIGHDANGER("You feel dizzy from the impact when [bound_xeno] slams into you!"))
+		A.sway_jitter(times = 5, steps = 3)
+		A.apply_effect(10, EYE_BLUR)
+
+
+/mob/living/carbon/xenomorph/lanky/handle_blood_splatter(splatter_dir)
+	new /obj/effect/temp_visual/dir_setting/bloodsplatter/yautjasplatter(loc, splatter_dir)
diff --git a/colonialmarines.dme b/colonialmarines.dme
index e2b70b1e956..135d175c256 100644
--- a/colonialmarines.dme
+++ b/colonialmarines.dme
@@ -2251,6 +2251,7 @@
 #include "code\modules\mob\living\carbon\xenomorph\ai\movement\facehugger.dm"
 #include "code\modules\mob\living\carbon\xenomorph\ai\movement\linger.dm"
 #include "code\modules\mob\living\carbon\xenomorph\ai\movement\lurking.dm"
+#include "code\modules\mob\living\carbon\xenomorph\castes\Bodyburster.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Boiler.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Burrower.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Carrier.dm"
@@ -2262,6 +2263,7 @@
 #include "code\modules\mob\living\carbon\xenomorph\castes\Hellhound.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Hivelord.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\king.dm"
+#include "code\modules\mob\living\carbon\xenomorph\castes\Lanky.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Larva.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\lesser_drone.dm"
 #include "code\modules\mob\living\carbon\xenomorph\castes\Lurker.dm"
diff --git a/icons/mob/xenos/bodyburster.dmi b/icons/mob/xenos/bodyburster.dmi
new file mode 100644
index 00000000000..878f91ee1aa
Binary files /dev/null and b/icons/mob/xenos/bodyburster.dmi differ
diff --git a/icons/mob/xenos/lanky.dmi b/icons/mob/xenos/lanky.dmi
new file mode 100644
index 00000000000..7363e0caec5
Binary files /dev/null and b/icons/mob/xenos/lanky.dmi differ
