diff --git a/code/modules/gear_presets/synths.dm b/code/modules/gear_presets/synths.dm
index ee178bbcb23..53b3338e7f8 100644
--- a/code/modules/gear_presets/synths.dm
+++ b/code/modules/gear_presets/synths.dm
@@ -1259,3 +1259,40 @@
 	name = "Synthetic - Security Android - Malfunction"
 	faction = FACTION_MALF_SYNTH
 	faction_group = list(FACTION_MALF_SYNTH)
+
+/datum/equipment_preset/synth/hybrid
+	name = "Xeno-Human Hybrid"
+	flags = EQUIPMENT_PRESET_EXTRA
+	faction = FACTION_COLONIST
+	assignment = JOB_COLONIST
+	rank = JOB_COLONIST
+	skills = /datum/skills/infiltrator_synthetic
+	idtype = /obj/item/card/id/lanyard
+	languages = list(LANGUAGE_ENGLISH, LANGUAGE_JAPANESE, LANGUAGE_SPANISH, LANGUAGE_XENOMORPH)
+	paygrades = list(PAY_SHORT_CIV = JOB_PLAYTIME_TIER_0)
+
+/datum/equipment_preset/synth/hybrid/New()
+	. = ..()
+	access = get_access(ACCESS_LIST_GLOBAL)
+
+/datum/equipment_preset/synth/hybrid/load_race(mob/living/carbon/human/new_human)
+	new_human.set_species("Hybrid")
+	new_human.h_style = pick("Undercut, Top", "CIA", "Mulder", "Pixie Cut Left", "Pixie Cut Right", "Scully", "Pvt. Redding", "Bun", "Short Bangs")
+
+/datum/equipment_preset/synth/hybrid/load_skills(mob/living/carbon/human/new_human)
+		new_human.set_skills(/datum/skills/infiltrator_synthetic)
+		new_human.allow_gun_usage = TRUE
+
+/datum/equipment_preset/synth/hybrid/load_gear(mob/living/carbon/human/new_human)
+	add_random_synth_infiltrator_equipment(new_human)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/backpack/satchel(new_human), WEAR_BACK)
+	new_human.equip_to_slot_or_del(new /obj/item/device/radio/headset(new_human), WEAR_L_EAR)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/pouch/first_responder(new_human), WEAR_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/stack/medical/advanced/bruise_pack/predator(new_human), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/stack/medical/advanced/bruise_pack/predator(new_human), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/stack/medical/advanced/ointment/predator(new_human), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/stack/medical/splint/nano(new_human), WEAR_IN_R_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/storage/pouch/general/large(new_human), WEAR_L_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/weapon/butterfly(new_human), WEAR_IN_L_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/device/flashlight(new_human), WEAR_IN_L_STORE)
+	new_human.equip_to_slot_or_del(new /obj/item/weapon/chloroform(new_human), WEAR_IN_L_STORE)
diff --git a/code/modules/mob/living/carbon/human/life/handle_pulse.dm b/code/modules/mob/living/carbon/human/life/handle_pulse.dm
index 11c07e253fd..7238ae88355 100644
--- a/code/modules/mob/living/carbon/human/life/handle_pulse.dm
+++ b/code/modules/mob/living/carbon/human/life/handle_pulse.dm
@@ -11,4 +11,7 @@
 	if(floor(blood_volume) <= BLOOD_VOLUME_BAD) //How much blood do we have
 		return PULSE_THREADY //not enough :(
 
+	if(species && species.mob_flags & NOBIOSCAN)
+		return pick(PULSE_THREADY, PULSE_SLOW, PULSE_FAST, PULSE_2FAST) //Scanner freaking the hell out
+
 	return PULSE_NORM
diff --git a/code/modules/mob/living/carbon/human/powers/human_powers.dm b/code/modules/mob/living/carbon/human/powers/human_powers.dm
index 8daf9a7488d..d0a147f2611 100644
--- a/code/modules/mob/living/carbon/human/powers/human_powers.dm
+++ b/code/modules/mob/living/carbon/human/powers/human_powers.dm
@@ -136,7 +136,7 @@
 		var/mob/living/M = G.grabbed_thing
 		if(!istype(M))
 			return //wut
-		M.apply_damage(50,BRUTE)
+		M.apply_damage(100,BRUTE)
 		if(M.stat == 2)
 			M.gib(create_cause_data("gutting", usr))
 
diff --git a/code/modules/mob/living/carbon/human/species/synthetic.dm b/code/modules/mob/living/carbon/human/species/synthetic.dm
index 64e97ffa156..5bc082f2cbb 100644
--- a/code/modules/mob/living/carbon/human/species/synthetic.dm
+++ b/code/modules/mob/living/carbon/human/species/synthetic.dm
@@ -133,3 +133,83 @@
 	bloodsplatter_type = /obj/effect/temp_visual/dir_setting/bloodsplatter/human
 
 	blood_color = BLOOD_COLOR_HUMAN
+
+
+/datum/species/synthetic/hybrid
+	group = SPECIES_HUMAN
+	name = "Hybrid"
+	name_plural = "Hybrids"
+	unarmed_type = /datum/unarmed_attack/punch/strong
+	secondary_unarmed_type = /datum/unarmed_attack/bite/strong
+	pain_type = /datum/pain/yautja
+	death_message = "doubles over, unleashes a horrible, ear-shattering scream, then falls motionless and still..."
+	death_sound = 'sound/voice/scream_horror1.ogg'
+
+	mob_inherent_traits = list(TRAIT_SUPER_STRONG, TRAIT_IRON_TEETH, TRAIT_DEXTROUS, TRAIT_FOREIGN_BIO)
+
+	bloodsplatter_type = /obj/effect/temp_visual/dir_setting/bloodsplatter/xenosplatter
+
+	total_health = 125 //Tougher than a human, but not quite full bug level
+
+	body_temperature = 250
+
+	mob_flags = KNOWS_TECHNOLOGY|NOBIOSCAN
+	flags = NO_BREATHE|NO_CLONE_LOSS|NO_POISON|NO_NEURO|NO_SHRAPNEL|SPECIAL_BONEBREAK|HAS_HARDCRIT|HAS_UNDERWEAR
+	flags_sight = SEE_MOBS
+	blood_color = "#550303"
+
+	has_organ = list(
+		"heart" = /datum/internal_organ/heart,
+		"lungs" = /datum/internal_organ/lungs,
+		"liver" = /datum/internal_organ/liver,
+		"kidneys" =  /datum/internal_organ/kidneys,
+		"brain" = /datum/internal_organ/brain,
+		"eyes" =  /datum/internal_organ/eyes
+		)
+
+	brute_mod = 0.35 // chitinous skin or something
+	burn_mod = 0.75
+	knock_down_reduction = 5
+	stun_reduction = 5
+	acid_blood_dodge_chance = 50
+	weed_slowdown_mult = 0 // no slowdown!
+	darksight = 5
+	default_lighting_alpha = LIGHTING_PLANE_ALPHA_YAUTJA
+
+	inherent_verbs = list(
+		/mob/living/carbon/human/synthetic/proc/toggle_HUD,
+		/mob/living/carbon/human/proc/toggle_inherent_nightvison,
+		/mob/living/carbon/human/proc/psychic_whisper,
+		/mob/living/carbon/human/proc/gut,
+		/mob/living/carbon/human/proc/commune,
+		/mob/living/carbon/human/proc/tackle,
+	)
+
+/datum/species/synthetic/hybrid/handle_on_fire(humanoidmob)
+	. = ..()
+	INVOKE_ASYNC(humanoidmob, TYPE_PROC_REF(/mob, emote), pick("pain", "scream"))
+
+/datum/species/synthetic/hybrid/handle_post_spawn(mob/living/carbon/human/hybrid)
+	. = ..()
+	GLOB.alive_human_list -= hybrid
+	hybrid.universal_understand = 1
+
+	hybrid.blood_type = "???"
+	for(var/obj/limb/limb in hybrid.limbs)
+		switch(limb.name)
+			if("groin","chest")
+				limb.min_broken_damage = 145
+				limb.max_damage = 150
+				limb.time_to_knit = 1200 // 2 minutes to self heal bone break, time is in tenths of a second to auto heal this
+			if("head")
+				limb.min_broken_damage = 140
+				limb.max_damage = 150
+				limb.time_to_knit = 600 // 1 minute to self heal bone break, time is in tenths of a second
+			if("l_hand","r_hand","r_foot","l_foot")
+				limb.min_broken_damage = 145
+				limb.max_damage = 150
+				limb.time_to_knit = 600 // 1 minute to self heal bone break, time is in tenths of a second
+			if("r_leg","r_arm","l_leg","l_arm")
+				limb.min_broken_damage = 145
+				limb.max_damage = 150
+				limb.time_to_knit = 600 // 1 minute to self heal bone break, time is in tenths of a second
